[*] Start mutation process:
   - targets: graph1.py
   - tests: ts-3-5/test_3-5_0-9_298.py
[*] 6 tests passed:
   - test_3-5_0-9_298 [0.18071 s]
[*] Start mutants generation and execution:
   - [#   1] AOD graph1: 
--------------------------------------------------------------------------------
   57:         visited = [0] * self.verticies
   58:         
   59:         def valid(node):
   60:             
-  61:             if visited[node] == -1:
+  61:             if visited[node] == 1:
   62:                 return False
   63:             elif visited[node] == 1:
   64:                 return True
   65:             visited[node] = -1
--------------------------------------------------------------------------------
[0.22178 s] killed by ts-3-5/test_3-5_0-9_298.py::test_graph_has_cycle

def test_graph_has_cycle():
        g = Graph(4)
        g.add_edge(0, 1)
        g.add_edge(1, 2)
        g.add_edge(2, 0)
>       assert g.has_cycle() == True

ts-3-5/test_3-5_0-9_298.py:23: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
graph1:77: in has_cycle
    ???
graph1:67: in valid
    ???
graph1:67: in valid
    ???
graph1:67: in valid
    ???
graph1:67: in valid
    ???
E   RecursionError: maximum recursion depth exceeded in comparison
!!! Recursion detected (same locals & position)
   - [#   2] AOD graph1: 
--------------------------------------------------------------------------------
   61:             if visited[node] == -1:
   62:                 return False
   63:             elif visited[node] == 1:
   64:                 return True
-  65:             visited[node] = -1
+  65:             visited[node] = 1
   66:             for neighbor in self.graph[node]:
   67:                 if not (valid(neighbor)):
   68:                     return False
   69:             return True
--------------------------------------------------------------------------------
[0.19879 s] killed by ts-3-5/test_3-5_0-9_298.py::test_graph_has_cycle

def test_graph_has_cycle():
        g = Graph(4)
        g.add_edge(0, 1)
        g.add_edge(1, 2)
        g.add_edge(2, 0)
>       assert g.has_cycle() == True

ts-3-5/test_3-5_0-9_298.py:23: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
graph1:77: in has_cycle
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

node = 3

>   ???
E   KeyError: 3

graph1:66: KeyError
   - [#   3] AOR graph1: 
--------------------------------------------------------------------------------
   53:             False if the graph is acyclic.
   54: 
   55:         '''
   56:         
-  57:         visited = [0] * self.verticies
+  57:         visited = [0] / self.verticies
   58:         
   59:         def valid(node):
   60:             
   61:             if visited[node] == -1:
--------------------------------------------------------------------------------
[0.19610 s] incompetent

TypeError: graph1:57: TypeError: unsupported operand type(s) for /: 'list' and 'int'

   - [#   4] AOR graph1: 
--------------------------------------------------------------------------------
   53:             False if the graph is acyclic.
   54: 
   55:         '''
   56:         
-  57:         visited = [0] * self.verticies
+  57:         visited = [0] // self.verticies
   58:         
   59:         def valid(node):
   60:             
   61:             if visited[node] == -1:
--------------------------------------------------------------------------------
[0.19426 s] incompetent

TypeError: graph1:57: TypeError: unsupported operand type(s) for //: 'list' and 'int'

   - [#   5] AOR graph1: 
--------------------------------------------------------------------------------
   53:             False if the graph is acyclic.
   54: 
   55:         '''
   56:         
-  57:         visited = [0] * self.verticies
+  57:         visited = [0] ** self.verticies
   58:         
   59:         def valid(node):
   60:             
   61:             if visited[node] == -1:
--------------------------------------------------------------------------------
[0.22230 s] incompetent

TypeError: graph1:57: TypeError: unsupported operand type(s) for ** or pow(): 'list' and 'int'

   - [#   6] AOR graph1: 
--------------------------------------------------------------------------------
   57:         visited = [0] * self.verticies
   58:         
   59:         def valid(node):
   60:             
-  61:             if visited[node] == -1:
+  61:             if visited[node] == +1:
   62:                 return False
   63:             elif visited[node] == 1:
   64:                 return True
   65:             visited[node] = -1
--------------------------------------------------------------------------------
[0.18571 s] killed by ts-3-5/test_3-5_0-9_298.py::test_graph_has_cycle

def test_graph_has_cycle():
        g = Graph(4)
        g.add_edge(0, 1)
        g.add_edge(1, 2)
        g.add_edge(2, 0)
>       assert g.has_cycle() == True

ts-3-5/test_3-5_0-9_298.py:23: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
graph1:77: in has_cycle
    ???
graph1:67: in valid
    ???
graph1:67: in valid
    ???
graph1:67: in valid
    ???
graph1:67: in valid
    ???
E   RecursionError: maximum recursion depth exceeded in comparison
!!! Recursion detected (same locals & position)
   - [#   7] AOR graph1: 
--------------------------------------------------------------------------------
   61:             if visited[node] == -1:
   62:                 return False
   63:             elif visited[node] == 1:
   64:                 return True
-  65:             visited[node] = -1
+  65:             visited[node] = +1
   66:             for neighbor in self.graph[node]:
   67:                 if not (valid(neighbor)):
   68:                     return False
   69:             return True
--------------------------------------------------------------------------------
[0.19563 s] killed by ts-3-5/test_3-5_0-9_298.py::test_graph_has_cycle

def test_graph_has_cycle():
        g = Graph(4)
        g.add_edge(0, 1)
        g.add_edge(1, 2)
        g.add_edge(2, 0)
>       assert g.has_cycle() == True

ts-3-5/test_3-5_0-9_298.py:23: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
graph1:77: in has_cycle
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

node = 3

>   ???
E   KeyError: 3

graph1:66: KeyError
   - [#   8] AOR graph1: 
--------------------------------------------------------------------------------
  127:         self.val = val
  128:         self.adjacent = {}
  129:     
  130:     def __str__(self):
- 131:         return (str(self.val) + ' adjacent: ') + str([x for x in self.adjacent])
+ 131:         return (str(self.val) - ' adjacent: ') + str([x for x in self.adjacent])
  132:     
  133:     def add_adjacent(self, node, weight=0):
  134:         self.adjacent[node] = weight
  135:     
--------------------------------------------------------------------------------
[0.17537 s] survived
   - [#   9] AOR graph1: 
--------------------------------------------------------------------------------
  127:         self.val = val
  128:         self.adjacent = {}
  129:     
  130:     def __str__(self):
- 131:         return (str(self.val) + ' adjacent: ') + str([x for x in self.adjacent])
+ 131:         return (str(self.val) + ' adjacent: ') - str([x for x in self.adjacent])
  132:     
  133:     def add_adjacent(self, node, weight=0):
  134:         self.adjacent[node] = weight
  135:     
--------------------------------------------------------------------------------
[0.17065 s] survived
   - [#  10] CDI graph1: 
--------------------------------------------------------------------------------
    1: from collections import defaultdict, deque
    2: 
    3: class GraphNode:
    4:     
-   5:     def __init__(self, val):
-   6:         self.val = val
-   7:         self.adjacent_list = set()
-   8:     
+   5:     @classmethod
+   6:     def __init__(self, val):
+   7:         self.val = val
+   8:         self.adjacent_list = set()
    9:     def add_adjacent(self, node):
   10:         self.adjacent_list.add(node)
   11:     
   12:     def remove_adjacent(self, node):
--------------------------------------------------------------------------------
[0.17416 s] survived
   - [#  11] CDI graph1: 
--------------------------------------------------------------------------------
    5:     def __init__(self, val):
    6:         self.val = val
    7:         self.adjacent_list = set()
    8:     
-   9:     def add_adjacent(self, node):
-  10:         self.adjacent_list.add(node)
-  11:     
+   9:     @classmethod
+  10:     def add_adjacent(self, node):
+  11:         self.adjacent_list.add(node)
   12:     def remove_adjacent(self, node):
   13:         self.adjacent_list.remove(node)
   14: 
   15: class Graph:
--------------------------------------------------------------------------------
[0.16874 s] survived
   - [#  12] CDI graph1: 
--------------------------------------------------------------------------------
    8:     
    9:     def add_adjacent(self, node):
   10:         self.adjacent_list.add(node)
   11:     
-  12:     def remove_adjacent(self, node):
-  13:         self.adjacent_list.remove(node)
-  14: 
+  12:     @classmethod
+  13:     def remove_adjacent(self, node):
+  14:         self.adjacent_list.remove(node)
   15: class Graph:
   16:     '''
   17:     A directed graph represented with an adjacency list.
   18:     '''
--------------------------------------------------------------------------------
[0.15780 s] survived
   - [#  13] CDI graph1: 
--------------------------------------------------------------------------------
   16:     '''
   17:     A directed graph represented with an adjacency list.
   18:     '''
   19:     
-  20:     def __init__(self, verticies):
-  21:         self.graph = {}
-  22:         self.verticies = verticies
-  23:     
+  20:     @classmethod
+  21:     def __init__(self, verticies):
+  22:         self.graph = {}
+  23:         self.verticies = verticies
   24:     def add_edge(self, source, destination):
   25:         '''
   26:         Add an edge to the graph.
   27: 
--------------------------------------------------------------------------------
[0.17210 s] survived
   - [#  14] CDI graph1: 
--------------------------------------------------------------------------------
   20:     def __init__(self, verticies):
   21:         self.graph = {}
   22:         self.verticies = verticies
   23:     
-  24:     def add_edge(self, source, destination):
-  25:         '''
-  26:         Add an edge to the graph.
-  27: 
-  28:         Add an edge pointing from source vertex
-  29:         to destination vertex.
-  30: 
-  31:         Args:
-  32:             source: the source vertex
-  33:             destination: the destination vertex
-  34: 
-  35:         '''
-  36:         if len(self.graph) > self.verticies:
-  37:             raise IndexError('Too many verticies in graph.')
-  38:         
-  39:         if source in self.graph:
-  40:             self.graph[source].append(destination)
-  41:         else:
-  42:             self.graph[source] = [destination]
-  43:         
-  44:         if destination not in self.graph:
-  45:             self.graph[destination] = []
-  46:     
+  24:     @classmethod
+  25:     def add_edge(self, source, destination):
+  26:         '''
+  27:         Add an edge to the graph.
+  28: 
+  29:         Add an edge pointing from source vertex
+  30:         to destination vertex.
+  31: 
+  32:         Args:
+  33:             source: the source vertex
+  34:             destination: the destination vertex
+  35: 
+  36:         '''
+  37:         if len(self.graph) > self.verticies:
+  38:             raise IndexError('Too many verticies in graph.')
+  39:         
+  40:         if source in self.graph:
+  41:             self.graph[source].append(destination)
+  42:         else:
+  43:             self.graph[source] = [destination]
+  44:         
+  45:         if destination not in self.graph:
+  46:             self.graph[destination] = []
   47:     def has_cycle(self):
   48:         '''
   49:         Detect if a graph has a cycle.
   50: 
--------------------------------------------------------------------------------
[0.19451 s] killed by ts-3-5/test_3-5_0-9_298.py::test_graph_add_edge

def test_graph_add_edge():
        g = Graph(4)
>       g.add_edge(0, 1)

ts-3-5/test_3-5_0-9_298.py:15: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <class 'graph1.Graph'>, source = 0, destination = 1

>   ???
E   AttributeError: type object 'Graph' has no attribute 'graph'

graph1:37: AttributeError
   - [#  15] CDI graph1: 
--------------------------------------------------------------------------------
   43:         
   44:         if destination not in self.graph:
   45:             self.graph[destination] = []
   46:     
-  47:     def has_cycle(self):
-  48:         '''
-  49:         Detect if a graph has a cycle.
-  50: 
-  51:         Returns:
-  52:             True if the graph has a cycle and
-  53:             False if the graph is acyclic.
-  54: 
-  55:         '''
-  56:         
-  57:         visited = [0] * self.verticies
-  58:         
-  59:         def valid(node):
-  60:             
-  61:             if visited[node] == -1:
-  62:                 return False
-  63:             elif visited[node] == 1:
-  64:                 return True
-  65:             visited[node] = -1
-  66:             for neighbor in self.graph[node]:
-  67:                 if not (valid(neighbor)):
-  68:                     return False
-  69:             return True
-  70:         
-  71:         for node in range(self.verticies):
-  72:             
+  47:     @classmethod
+  48:     def has_cycle(self):
+  49:         '''
+  50:         Detect if a graph has a cycle.
+  51: 
+  52:         Returns:
+  53:             True if the graph has a cycle and
+  54:             False if the graph is acyclic.
+  55: 
+  56:         '''
+  57:         
+  58:         visited = [0] * self.verticies
+  59:         
+  60:         def valid(node):
+  61:             
+  62:             if visited[node] == -1:
+  63:                 return False
+  64:             elif visited[node] == 1:
+  65:                 return True
+  66:             visited[node] = -1
+  67:             for neighbor in self.graph[node]:
+  68:                 if not (valid(neighbor)):
+  69:                     return False
+  70:             return True
+  71:         
+  72:         for node in range(self.verticies):
   73:             
   74:             
-  75:             if not (visited[node]):
-  76:                 
-  77:                 if valid(node):
-  78:                     
-  79:                     visited = list(map((lambda x: (abs(x))), visited))
-  80:                 else:
-  81:                     
-  82:                     return True
-  83:         
-  84:         return False
-  85:     
+  75:             
+  76:             if not (visited[node]):
+  77:                 
+  78:                 if valid(node):
+  79:                     
+  80:                     visited = list(map((lambda x: (abs(x))), visited))
+  81:                 else:
+  82:                     
+  83:                     return True
+  84:         
+  85:         return False
   86:     def topological_sort(self):
   87:         """
   88:         Sort the graph topologically.
   89: 
--------------------------------------------------------------------------------
[0.19787 s] killed by ts-3-5/test_3-5_0-9_298.py::test_graph_has_cycle

def test_graph_has_cycle():
        g = Graph(4)
        g.add_edge(0, 1)
        g.add_edge(1, 2)
        g.add_edge(2, 0)
>       assert g.has_cycle() == True

ts-3-5/test_3-5_0-9_298.py:23: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <class 'graph1.Graph'>

>   ???
E   AttributeError: type object 'Graph' has no attribute 'verticies'

graph1:58: AttributeError
   - [#  16] CDI graph1: 
--------------------------------------------------------------------------------
   82:                     return True
   83:         
   84:         return False
   85:     
-  86:     def topological_sort(self):
-  87:         """
-  88:         Sort the graph topologically.
-  89: 
-  90:         A topological sort lists nodes in such a way
-  91:         that every node 's' in 's' -> 'd' directed pairs
-  92:         is listed before 'd.'  This will not work in a 
-  93:         graph that contains cycles.
-  94: 
-  95:         The algorithm looks at every node, and does a
-  96:         dfs for each node adjacent to the node and then adds
-  97:         the originating node to a stack once all adjacent
-  98:         nodes have been searched.  In the end, the stack
-  99:         will be in order of a possible topological sort.
- 100: 
- 101:         Topological sorts are not necessarily unique.
- 102: 
- 103:         Returns:
- 104:             A list of vertices in a topological ordering.
- 105: 
- 106:         """
- 107:         visited = set()
- 108:         stack = deque()
- 109:         
- 110:         def dfs(vertex):
- 111:             visited.add(vertex)
- 112:             for j in self.graph[vertex]:
- 113:                 if j not in visited:
- 114:                     dfs(j)
- 115:             
- 116:             stack.appendleft(vertex)
- 117:         
- 118:         for (key, _) in self.graph.items():
- 119:             if key not in visited:
- 120:                 dfs(key)
- 121:         
- 122:         return list(stack)
- 123: 
+  86:     @classmethod
+  87:     def topological_sort(self):
+  88:         """
+  89:         Sort the graph topologically.
+  90: 
+  91:         A topological sort lists nodes in such a way
+  92:         that every node 's' in 's' -> 'd' directed pairs
+  93:         is listed before 'd.'  This will not work in a 
+  94:         graph that contains cycles.
+  95: 
+  96:         The algorithm looks at every node, and does a
+  97:         dfs for each node adjacent to the node and then adds
+  98:         the originating node to a stack once all adjacent
+  99:         nodes have been searched.  In the end, the stack
+ 100:         will be in order of a possible topological sort.
+ 101: 
+ 102:         Topological sorts are not necessarily unique.
+ 103: 
+ 104:         Returns:
+ 105:             A list of vertices in a topological ordering.
+ 106: 
+ 107:         """
+ 108:         visited = set()
+ 109:         stack = deque()
+ 110:         
+ 111:         def dfs(vertex):
+ 112:             visited.add(vertex)
+ 113:             for j in self.graph[vertex]:
+ 114:                 if j not in visited:
+ 115:                     dfs(j)
+ 116:             
+ 117:             stack.appendleft(vertex)
+ 118:         
+ 119:         for (key, _) in self.graph.items():
+ 120:             if key not in visited:
+ 121:                 dfs(key)
+ 122:         
+ 123:         return list(stack)
  124: class WeightedGraphNode:
  125:     
  126:     def __init__(self, val):
  127:         self.val = val
--------------------------------------------------------------------------------
[0.22334 s] killed by ts-3-5/test_3-5_0-9_298.py::test_graph_topological_sort

def test_graph_topological_sort():
        g = Graph(4)
        g.add_edge(0, 1)
        g.add_edge(0, 2)
        g.add_edge(1, 3)
        g.add_edge(2, 3)
>       assert g.topological_sort() == [0, 2, 1, 3]

ts-3-5/test_3-5_0-9_298.py:31: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <class 'graph1.Graph'>

>   ???
E   AttributeError: type object 'Graph' has no attribute 'graph'

graph1:119: AttributeError
   - [#  17] CDI graph1: 
--------------------------------------------------------------------------------
  122:         return list(stack)
  123: 
  124: class WeightedGraphNode:
  125:     
- 126:     def __init__(self, val):
- 127:         self.val = val
- 128:         self.adjacent = {}
- 129:     
+ 126:     @classmethod
+ 127:     def __init__(self, val):
+ 128:         self.val = val
+ 129:         self.adjacent = {}
  130:     def __str__(self):
  131:         return (str(self.val) + ' adjacent: ') + str([x for x in self.adjacent])
  132:     
  133:     def add_adjacent(self, node, weight=0):
--------------------------------------------------------------------------------
[0.16327 s] survived
   - [#  18] CDI graph1: 
--------------------------------------------------------------------------------
  126:     def __init__(self, val):
  127:         self.val = val
  128:         self.adjacent = {}
  129:     
- 130:     def __str__(self):
- 131:         return (str(self.val) + ' adjacent: ') + str([x for x in self.adjacent])
- 132:     
+ 130:     @classmethod
+ 131:     def __str__(self):
+ 132:         return (str(self.val) + ' adjacent: ') + str([x for x in self.adjacent])
  133:     def add_adjacent(self, node, weight=0):
  134:         self.adjacent[node] = weight
  135:     
  136:     def remove_adjacent(self, node):
--------------------------------------------------------------------------------
[0.17443 s] survived
   - [#  19] CDI graph1: 
--------------------------------------------------------------------------------
  129:     
  130:     def __str__(self):
  131:         return (str(self.val) + ' adjacent: ') + str([x for x in self.adjacent])
  132:     
- 133:     def add_adjacent(self, node, weight=0):
- 134:         self.adjacent[node] = weight
- 135:     
+ 133:     @classmethod
+ 134:     def add_adjacent(self, node, weight=0):
+ 135:         self.adjacent[node] = weight
  136:     def remove_adjacent(self, node):
  137:         del self.adjacent[node]
  138: 
  139: class WeightedGraph:
--------------------------------------------------------------------------------
[0.19639 s] killed by ts-3-5/test_3-5_0-9_298.py::test_weighted_graph_add_edge

def test_weighted_graph_add_edge():
        wg = WeightedGraph()
>       wg.add_edge(0, 1, 5)

ts-3-5/test_3-5_0-9_298.py:36: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
graph1:156: in add_edge
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <class 'graph1.WeightedGraphNode'>, node = 1, weight = 5

>   ???
E   AttributeError: type object 'WeightedGraphNode' has no attribute 'adjacent'

graph1:135: AttributeError
   - [#  20] CDI graph1: 
--------------------------------------------------------------------------------
  132:     
  133:     def add_adjacent(self, node, weight=0):
  134:         self.adjacent[node] = weight
  135:     
- 136:     def remove_adjacent(self, node):
- 137:         del self.adjacent[node]
- 138: 
+ 136:     @classmethod
+ 137:     def remove_adjacent(self, node):
+ 138:         del self.adjacent[node]
  139: class WeightedGraph:
  140:     
  141:     def __init__(self):
  142:         self.vertices = {}
--------------------------------------------------------------------------------
[0.19791 s] killed by ts-3-5/test_3-5_0-9_298.py::test_weighted_graph_remove_edge

def test_weighted_graph_remove_edge():
        wg = WeightedGraph()
        wg.add_edge(0, 1, 5)
>       wg.remove_edge(0, 1)

ts-3-5/test_3-5_0-9_298.py:42: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
graph1:164: in remove_edge
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <class 'graph1.WeightedGraphNode'>, node = 1

>   ???
E   AttributeError: type object 'WeightedGraphNode' has no attribute 'adjacent'

graph1:138: AttributeError
   - [#  21] CDI graph1: 
--------------------------------------------------------------------------------
  137:         del self.adjacent[node]
  138: 
  139: class WeightedGraph:
  140:     
- 141:     def __init__(self):
- 142:         self.vertices = {}
- 143:     
+ 141:     @classmethod
+ 142:     def __init__(self):
+ 143:         self.vertices = {}
  144:     def __str__(self):
  145:         return ', '.join([node.__str__() for node in self.vertices.values()])
  146:     
  147:     def add_vertex(self, val):
--------------------------------------------------------------------------------
[0.17184 s] survived
   - [#  22] CDI graph1: 
--------------------------------------------------------------------------------
  140:     
  141:     def __init__(self):
  142:         self.vertices = {}
  143:     
- 144:     def __str__(self):
- 145:         return ', '.join([node.__str__() for node in self.vertices.values()])
- 146:     
+ 144:     @classmethod
+ 145:     def __str__(self):
+ 146:         return ', '.join([node.__str__() for node in self.vertices.values()])
  147:     def add_vertex(self, val):
  148:         vertex = WeightedGraphNode(val)
  149:         self.vertices[val] = vertex
  150:     
--------------------------------------------------------------------------------
[0.16789 s] survived
   - [#  23] CDI graph1: 
--------------------------------------------------------------------------------
  143:     
  144:     def __str__(self):
  145:         return ', '.join([node.__str__() for node in self.vertices.values()])
  146:     
- 147:     def add_vertex(self, val):
- 148:         vertex = WeightedGraphNode(val)
- 149:         self.vertices[val] = vertex
- 150:     
+ 147:     @classmethod
+ 148:     def add_vertex(self, val):
+ 149:         vertex = WeightedGraphNode(val)
+ 150:         self.vertices[val] = vertex
  151:     def add_edge(self, src, dst, weight=0):
  152:         if src not in self.vertices:
  153:             self.add_vertex(src)
  154:         if dst not in self.vertices:
--------------------------------------------------------------------------------
[0.18118 s] killed by ts-3-5/test_3-5_0-9_298.py::test_weighted_graph_add_edge

def test_weighted_graph_add_edge():
        wg = WeightedGraph()
>       wg.add_edge(0, 1, 5)

ts-3-5/test_3-5_0-9_298.py:36: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
graph1:153: in add_edge
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <class 'graph1.WeightedGraph'>, val = 0

>   ???
E   AttributeError: type object 'WeightedGraph' has no attribute 'vertices'

graph1:150: AttributeError
   - [#  24] CDI graph1: 
--------------------------------------------------------------------------------
  147:     def add_vertex(self, val):
  148:         vertex = WeightedGraphNode(val)
  149:         self.vertices[val] = vertex
  150:     
- 151:     def add_edge(self, src, dst, weight=0):
- 152:         if src not in self.vertices:
- 153:             self.add_vertex(src)
- 154:         if dst not in self.vertices:
- 155:             self.add_vertex(dst)
- 156:         self.vertices[src].add_adjacent(dst, weight)
- 157:     
+ 151:     @classmethod
+ 152:     def add_edge(self, src, dst, weight=0):
+ 153:         if src not in self.vertices:
+ 154:             self.add_vertex(src)
+ 155:         if dst not in self.vertices:
+ 156:             self.add_vertex(dst)
+ 157:         self.vertices[src].add_adjacent(dst, weight)
  158:     def remove_edge(self, src, dst):
  159:         if src not in self.vertices:
  160:             return
  161:         if dst not in self.vertices:
--------------------------------------------------------------------------------
[0.19930 s] killed by ts-3-5/test_3-5_0-9_298.py::test_weighted_graph_add_edge

def test_weighted_graph_add_edge():
        wg = WeightedGraph()
>       wg.add_edge(0, 1, 5)

ts-3-5/test_3-5_0-9_298.py:36: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <class 'graph1.WeightedGraph'>, src = 0, dst = 1, weight = 5

>   ???
E   AttributeError: type object 'WeightedGraph' has no attribute 'vertices'

graph1:153: AttributeError
   - [#  25] CDI graph1: 
--------------------------------------------------------------------------------
  154:         if dst not in self.vertices:
  155:             self.add_vertex(dst)
  156:         self.vertices[src].add_adjacent(dst, weight)
  157:     
- 158:     def remove_edge(self, src, dst):
- 159:         if src not in self.vertices:
- 160:             return
- 161:         if dst not in self.vertices:
- 162:             return
- 163:         if dst in self.vertices[src].adjacent:
- 164:             self.vertices[src].remove_adjacent(dst)
+ 158:     @classmethod
+ 159:     def remove_edge(self, src, dst):
+ 160:         if src not in self.vertices:
+ 161:             return
+ 162:         if dst not in self.vertices:
+ 163:             return
+ 164:         if dst in self.vertices[src].adjacent:
+ 165:             self.vertices[src].remove_adjacent(dst)
--------------------------------------------------------------------------------
[0.21509 s] killed by ts-3-5/test_3-5_0-9_298.py::test_weighted_graph_remove_edge

def test_weighted_graph_remove_edge():
        wg = WeightedGraph()
        wg.add_edge(0, 1, 5)
>       wg.remove_edge(0, 1)

ts-3-5/test_3-5_0-9_298.py:42: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <class 'graph1.WeightedGraph'>, src = 0, dst = 1

>   ???
E   AttributeError: type object 'WeightedGraph' has no attribute 'vertices'

graph1:160: AttributeError
   - [#  26] COD graph1: 
--------------------------------------------------------------------------------
   40:             self.graph[source].append(destination)
   41:         else:
   42:             self.graph[source] = [destination]
   43:         
-  44:         if destination not in self.graph:
+  44:         if destination in self.graph:
   45:             self.graph[destination] = []
   46:     
   47:     def has_cycle(self):
   48:         '''
--------------------------------------------------------------------------------
[0.19417 s] killed by ts-3-5/test_3-5_0-9_298.py::test_graph_add_edge

def test_graph_add_edge():
        g = Graph(4)
        g.add_edge(0, 1)
>       assert g.graph == {0: [1], 1: []}
E       AssertionError

ts-3-5/test_3-5_0-9_298.py:16: AssertionError
   - [#  27] COD graph1: 
--------------------------------------------------------------------------------
   63:             elif visited[node] == 1:
   64:                 return True
   65:             visited[node] = -1
   66:             for neighbor in self.graph[node]:
-  67:                 if not (valid(neighbor)):
+  67:                 if valid(neighbor):
   68:                     return False
   69:             return True
   70:         
   71:         for node in range(self.verticies):
--------------------------------------------------------------------------------
[0.20365 s] killed by ts-3-5/test_3-5_0-9_298.py::test_graph_has_cycle

def test_graph_has_cycle():
        g = Graph(4)
        g.add_edge(0, 1)
        g.add_edge(1, 2)
        g.add_edge(2, 0)
>       assert g.has_cycle() == True

ts-3-5/test_3-5_0-9_298.py:23: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
graph1:77: in has_cycle
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

node = 3

>   ???
E   KeyError: 3

graph1:66: KeyError
   - [#  28] COD graph1: 
--------------------------------------------------------------------------------
   71:         for node in range(self.verticies):
   72:             
   73:             
   74:             
-  75:             if not (visited[node]):
+  75:             if visited[node]:
   76:                 
   77:                 if valid(node):
   78:                     
   79:                     visited = list(map((lambda x: (abs(x))), visited))
--------------------------------------------------------------------------------
[0.18310 s] killed by ts-3-5/test_3-5_0-9_298.py::test_graph_has_cycle

def test_graph_has_cycle():
        g = Graph(4)
        g.add_edge(0, 1)
        g.add_edge(1, 2)
        g.add_edge(2, 0)
>       assert g.has_cycle() == True
E       AssertionError

ts-3-5/test_3-5_0-9_298.py:23: AssertionError
   - [#  29] COD graph1: 
--------------------------------------------------------------------------------
  109:         
  110:         def dfs(vertex):
  111:             visited.add(vertex)
  112:             for j in self.graph[vertex]:
- 113:                 if j not in visited:
+ 113:                 if j in visited:
  114:                     dfs(j)
  115:             
  116:             stack.appendleft(vertex)
  117:         
--------------------------------------------------------------------------------
[0.19938 s] killed by ts-3-5/test_3-5_0-9_298.py::test_graph_topological_sort

def test_graph_topological_sort():
        g = Graph(4)
        g.add_edge(0, 1)
        g.add_edge(0, 2)
        g.add_edge(1, 3)
        g.add_edge(2, 3)
>       assert g.topological_sort() == [0, 2, 1, 3]
E       AssertionError

ts-3-5/test_3-5_0-9_298.py:31: AssertionError
   - [#  30] COD graph1: 
--------------------------------------------------------------------------------
  115:             
  116:             stack.appendleft(vertex)
  117:         
  118:         for (key, _) in self.graph.items():
- 119:             if key not in visited:
+ 119:             if key in visited:
  120:                 dfs(key)
  121:         
  122:         return list(stack)
  123: 
--------------------------------------------------------------------------------
[0.19711 s] killed by ts-3-5/test_3-5_0-9_298.py::test_graph_topological_sort

def test_graph_topological_sort():
        g = Graph(4)
        g.add_edge(0, 1)
        g.add_edge(0, 2)
        g.add_edge(1, 3)
        g.add_edge(2, 3)
>       assert g.topological_sort() == [0, 2, 1, 3]
E       AssertionError

ts-3-5/test_3-5_0-9_298.py:31: AssertionError
   - [#  31] COD graph1: 
--------------------------------------------------------------------------------
  148:         vertex = WeightedGraphNode(val)
  149:         self.vertices[val] = vertex
  150:     
  151:     def add_edge(self, src, dst, weight=0):
- 152:         if src not in self.vertices:
+ 152:         if src in self.vertices:
  153:             self.add_vertex(src)
  154:         if dst not in self.vertices:
  155:             self.add_vertex(dst)
  156:         self.vertices[src].add_adjacent(dst, weight)
--------------------------------------------------------------------------------
[0.21611 s] killed by ts-3-5/test_3-5_0-9_298.py::test_weighted_graph_add_edge

def test_weighted_graph_add_edge():
        wg = WeightedGraph()
>       wg.add_edge(0, 1, 5)

ts-3-5/test_3-5_0-9_298.py:36: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph1.WeightedGraph object at 0x7026489de920>, src = 0, dst = 1
weight = 5

>   ???
E   KeyError: 0

graph1:156: KeyError
   - [#  32] COD graph1: 
--------------------------------------------------------------------------------
  150:     
  151:     def add_edge(self, src, dst, weight=0):
  152:         if src not in self.vertices:
  153:             self.add_vertex(src)
- 154:         if dst not in self.vertices:
+ 154:         if dst in self.vertices:
  155:             self.add_vertex(dst)
  156:         self.vertices[src].add_adjacent(dst, weight)
  157:     
  158:     def remove_edge(self, src, dst):
--------------------------------------------------------------------------------
[0.20428 s] killed by ts-3-5/test_3-5_0-9_298.py::test_weighted_graph_remove_edge

def test_weighted_graph_remove_edge():
        wg = WeightedGraph()
        wg.add_edge(0, 1, 5)
        wg.remove_edge(0, 1)
>       assert 1 not in wg.vertices[0].adjacent
E       AssertionError

ts-3-5/test_3-5_0-9_298.py:43: AssertionError
   - [#  33] COD graph1: 
--------------------------------------------------------------------------------
  155:             self.add_vertex(dst)
  156:         self.vertices[src].add_adjacent(dst, weight)
  157:     
  158:     def remove_edge(self, src, dst):
- 159:         if src not in self.vertices:
+ 159:         if src in self.vertices:
  160:             return
  161:         if dst not in self.vertices:
  162:             return
  163:         if dst in self.vertices[src].adjacent:
--------------------------------------------------------------------------------
[0.18339 s] killed by ts-3-5/test_3-5_0-9_298.py::test_weighted_graph_remove_edge

def test_weighted_graph_remove_edge():
        wg = WeightedGraph()
        wg.add_edge(0, 1, 5)
        wg.remove_edge(0, 1)
>       assert 1 not in wg.vertices[0].adjacent
E       AssertionError

ts-3-5/test_3-5_0-9_298.py:43: AssertionError
   - [#  34] COD graph1: 
--------------------------------------------------------------------------------
  157:     
  158:     def remove_edge(self, src, dst):
  159:         if src not in self.vertices:
  160:             return
- 161:         if dst not in self.vertices:
+ 161:         if dst in self.vertices:
  162:             return
  163:         if dst in self.vertices[src].adjacent:
  164:             self.vertices[src].remove_adjacent(dst)
--------------------------------------------------------------------------------
[0.19895 s] killed by ts-3-5/test_3-5_0-9_298.py::test_weighted_graph_remove_edge

def test_weighted_graph_remove_edge():
        wg = WeightedGraph()
        wg.add_edge(0, 1, 5)
        wg.remove_edge(0, 1)
>       assert 1 not in wg.vertices[0].adjacent
E       AssertionError

ts-3-5/test_3-5_0-9_298.py:43: AssertionError
   - [#  35] COI graph1: 
--------------------------------------------------------------------------------
   32:             source: the source vertex
   33:             destination: the destination vertex
   34: 
   35:         '''
-  36:         if len(self.graph) > self.verticies:
+  36:         if not (len(self.graph) > self.verticies):
   37:             raise IndexError('Too many verticies in graph.')
   38:         
   39:         if source in self.graph:
   40:             self.graph[source].append(destination)
--------------------------------------------------------------------------------
[0.19392 s] killed by ts-3-5/test_3-5_0-9_298.py::test_graph_add_edge

def test_graph_add_edge():
        g = Graph(4)
>       g.add_edge(0, 1)

ts-3-5/test_3-5_0-9_298.py:15: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph1.Graph object at 0x702648a7ee00>, source = 0, destination = 1

>   ???
E   IndexError: Too many verticies in graph.

graph1:37: IndexError
   - [#  36] COI graph1: 
--------------------------------------------------------------------------------
   35:         '''
   36:         if len(self.graph) > self.verticies:
   37:             raise IndexError('Too many verticies in graph.')
   38:         
-  39:         if source in self.graph:
+  39:         if not (source in self.graph):
   40:             self.graph[source].append(destination)
   41:         else:
   42:             self.graph[source] = [destination]
   43:         
--------------------------------------------------------------------------------
[0.19137 s] killed by ts-3-5/test_3-5_0-9_298.py::test_graph_add_edge

def test_graph_add_edge():
        g = Graph(4)
>       g.add_edge(0, 1)

ts-3-5/test_3-5_0-9_298.py:15: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph1.Graph object at 0x702648bfc3a0>, source = 0, destination = 1

>   ???
E   KeyError: 0

graph1:40: KeyError
   - [#  37] COI graph1: 
--------------------------------------------------------------------------------
   35:         '''
   36:         if len(self.graph) > self.verticies:
   37:             raise IndexError('Too many verticies in graph.')
   38:         
-  39:         if source in self.graph:
+  39:         if source not in self.graph:
   40:             self.graph[source].append(destination)
   41:         else:
   42:             self.graph[source] = [destination]
   43:         
--------------------------------------------------------------------------------
[0.19979 s] killed by ts-3-5/test_3-5_0-9_298.py::test_graph_add_edge

def test_graph_add_edge():
        g = Graph(4)
>       g.add_edge(0, 1)

ts-3-5/test_3-5_0-9_298.py:15: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph1.Graph object at 0x702648b37f10>, source = 0, destination = 1

>   ???
E   KeyError: 0

graph1:40: KeyError
   - [#  38] COI graph1: 
--------------------------------------------------------------------------------
   40:             self.graph[source].append(destination)
   41:         else:
   42:             self.graph[source] = [destination]
   43:         
-  44:         if destination not in self.graph:
+  44:         if not (destination not in self.graph):
   45:             self.graph[destination] = []
   46:     
   47:     def has_cycle(self):
   48:         '''
--------------------------------------------------------------------------------
[0.21012 s] killed by ts-3-5/test_3-5_0-9_298.py::test_graph_add_edge

def test_graph_add_edge():
        g = Graph(4)
        g.add_edge(0, 1)
>       assert g.graph == {0: [1], 1: []}
E       AssertionError

ts-3-5/test_3-5_0-9_298.py:16: AssertionError
   - [#  39] COI graph1: 
--------------------------------------------------------------------------------
   57:         visited = [0] * self.verticies
   58:         
   59:         def valid(node):
   60:             
-  61:             if visited[node] == -1:
+  61:             if not (visited[node] == -1):
   62:                 return False
   63:             elif visited[node] == 1:
   64:                 return True
   65:             visited[node] = -1
--------------------------------------------------------------------------------
[0.22151 s] survived
   - [#  40] COI graph1: 
--------------------------------------------------------------------------------
   59:         def valid(node):
   60:             
   61:             if visited[node] == -1:
   62:                 return False
-  63:             elif visited[node] == 1:
+  63:             elif not (visited[node] == 1):
   64:                 return True
   65:             visited[node] = -1
   66:             for neighbor in self.graph[node]:
   67:                 if not (valid(neighbor)):
--------------------------------------------------------------------------------
[0.19848 s] killed by ts-3-5/test_3-5_0-9_298.py::test_graph_has_cycle

def test_graph_has_cycle():
        g = Graph(4)
        g.add_edge(0, 1)
        g.add_edge(1, 2)
        g.add_edge(2, 0)
>       assert g.has_cycle() == True
E       AssertionError

ts-3-5/test_3-5_0-9_298.py:23: AssertionError
   - [#  41] COI graph1: 
--------------------------------------------------------------------------------
   63:             elif visited[node] == 1:
   64:                 return True
   65:             visited[node] = -1
   66:             for neighbor in self.graph[node]:
-  67:                 if not (valid(neighbor)):
+  67:                 if not ((not (valid(neighbor)))):
   68:                     return False
   69:             return True
   70:         
   71:         for node in range(self.verticies):
--------------------------------------------------------------------------------
[0.19804 s] killed by ts-3-5/test_3-5_0-9_298.py::test_graph_has_cycle

def test_graph_has_cycle():
        g = Graph(4)
        g.add_edge(0, 1)
        g.add_edge(1, 2)
        g.add_edge(2, 0)
>       assert g.has_cycle() == True

ts-3-5/test_3-5_0-9_298.py:23: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
graph1:77: in has_cycle
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

node = 3

>   ???
E   KeyError: 3

graph1:66: KeyError
   - [#  42] COI graph1: 
--------------------------------------------------------------------------------
   71:         for node in range(self.verticies):
   72:             
   73:             
   74:             
-  75:             if not (visited[node]):
+  75:             if not ((not (visited[node]))):
   76:                 
   77:                 if valid(node):
   78:                     
   79:                     visited = list(map((lambda x: (abs(x))), visited))
--------------------------------------------------------------------------------
[0.19411 s] killed by ts-3-5/test_3-5_0-9_298.py::test_graph_has_cycle

def test_graph_has_cycle():
        g = Graph(4)
        g.add_edge(0, 1)
        g.add_edge(1, 2)
        g.add_edge(2, 0)
>       assert g.has_cycle() == True
E       AssertionError

ts-3-5/test_3-5_0-9_298.py:23: AssertionError
   - [#  43] COI graph1: 
--------------------------------------------------------------------------------
   73:             
   74:             
   75:             if not (visited[node]):
   76:                 
-  77:                 if valid(node):
+  77:                 if not (valid(node)):
   78:                     
   79:                     visited = list(map((lambda x: (abs(x))), visited))
   80:                 else:
   81:                     
--------------------------------------------------------------------------------
[0.22369 s] killed by ts-3-5/test_3-5_0-9_298.py::test_graph_has_cycle

def test_graph_has_cycle():
        g = Graph(4)
        g.add_edge(0, 1)
        g.add_edge(1, 2)
        g.add_edge(2, 0)
>       assert g.has_cycle() == True

ts-3-5/test_3-5_0-9_298.py:23: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
graph1:77: in has_cycle
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

node = 3

>   ???
E   KeyError: 3

graph1:66: KeyError
   - [#  44] COI graph1: 
--------------------------------------------------------------------------------
  109:         
  110:         def dfs(vertex):
  111:             visited.add(vertex)
  112:             for j in self.graph[vertex]:
- 113:                 if j not in visited:
+ 113:                 if not (j not in visited):
  114:                     dfs(j)
  115:             
  116:             stack.appendleft(vertex)
  117:         
--------------------------------------------------------------------------------
[0.18117 s] killed by ts-3-5/test_3-5_0-9_298.py::test_graph_topological_sort

def test_graph_topological_sort():
        g = Graph(4)
        g.add_edge(0, 1)
        g.add_edge(0, 2)
        g.add_edge(1, 3)
        g.add_edge(2, 3)
>       assert g.topological_sort() == [0, 2, 1, 3]
E       AssertionError

ts-3-5/test_3-5_0-9_298.py:31: AssertionError
   - [#  45] COI graph1: 
--------------------------------------------------------------------------------
  115:             
  116:             stack.appendleft(vertex)
  117:         
  118:         for (key, _) in self.graph.items():
- 119:             if key not in visited:
+ 119:             if not (key not in visited):
  120:                 dfs(key)
  121:         
  122:         return list(stack)
  123: 
--------------------------------------------------------------------------------
[0.19963 s] killed by ts-3-5/test_3-5_0-9_298.py::test_graph_topological_sort

def test_graph_topological_sort():
        g = Graph(4)
        g.add_edge(0, 1)
        g.add_edge(0, 2)
        g.add_edge(1, 3)
        g.add_edge(2, 3)
>       assert g.topological_sort() == [0, 2, 1, 3]
E       AssertionError

ts-3-5/test_3-5_0-9_298.py:31: AssertionError
   - [#  46] COI graph1: 
--------------------------------------------------------------------------------
  148:         vertex = WeightedGraphNode(val)
  149:         self.vertices[val] = vertex
  150:     
  151:     def add_edge(self, src, dst, weight=0):
- 152:         if src not in self.vertices:
+ 152:         if not (src not in self.vertices):
  153:             self.add_vertex(src)
  154:         if dst not in self.vertices:
  155:             self.add_vertex(dst)
  156:         self.vertices[src].add_adjacent(dst, weight)
--------------------------------------------------------------------------------
[0.19397 s] killed by ts-3-5/test_3-5_0-9_298.py::test_weighted_graph_add_edge

def test_weighted_graph_add_edge():
        wg = WeightedGraph()
>       wg.add_edge(0, 1, 5)

ts-3-5/test_3-5_0-9_298.py:36: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph1.WeightedGraph object at 0x702648871930>, src = 0, dst = 1
weight = 5

>   ???
E   KeyError: 0

graph1:156: KeyError
   - [#  47] COI graph1: 
--------------------------------------------------------------------------------
  150:     
  151:     def add_edge(self, src, dst, weight=0):
  152:         if src not in self.vertices:
  153:             self.add_vertex(src)
- 154:         if dst not in self.vertices:
+ 154:         if not (dst not in self.vertices):
  155:             self.add_vertex(dst)
  156:         self.vertices[src].add_adjacent(dst, weight)
  157:     
  158:     def remove_edge(self, src, dst):
--------------------------------------------------------------------------------
[0.19742 s] killed by ts-3-5/test_3-5_0-9_298.py::test_weighted_graph_remove_edge

def test_weighted_graph_remove_edge():
        wg = WeightedGraph()
        wg.add_edge(0, 1, 5)
        wg.remove_edge(0, 1)
>       assert 1 not in wg.vertices[0].adjacent
E       AssertionError

ts-3-5/test_3-5_0-9_298.py:43: AssertionError
   - [#  48] COI graph1: 
--------------------------------------------------------------------------------
  155:             self.add_vertex(dst)
  156:         self.vertices[src].add_adjacent(dst, weight)
  157:     
  158:     def remove_edge(self, src, dst):
- 159:         if src not in self.vertices:
+ 159:         if not (src not in self.vertices):
  160:             return
  161:         if dst not in self.vertices:
  162:             return
  163:         if dst in self.vertices[src].adjacent:
--------------------------------------------------------------------------------
[0.20285 s] killed by ts-3-5/test_3-5_0-9_298.py::test_weighted_graph_remove_edge

def test_weighted_graph_remove_edge():
        wg = WeightedGraph()
        wg.add_edge(0, 1, 5)
        wg.remove_edge(0, 1)
>       assert 1 not in wg.vertices[0].adjacent
E       AssertionError

ts-3-5/test_3-5_0-9_298.py:43: AssertionError
   - [#  49] COI graph1: 
--------------------------------------------------------------------------------
  157:     
  158:     def remove_edge(self, src, dst):
  159:         if src not in self.vertices:
  160:             return
- 161:         if dst not in self.vertices:
+ 161:         if not (dst not in self.vertices):
  162:             return
  163:         if dst in self.vertices[src].adjacent:
  164:             self.vertices[src].remove_adjacent(dst)
--------------------------------------------------------------------------------
[0.18603 s] killed by ts-3-5/test_3-5_0-9_298.py::test_weighted_graph_remove_edge

def test_weighted_graph_remove_edge():
        wg = WeightedGraph()
        wg.add_edge(0, 1, 5)
        wg.remove_edge(0, 1)
>       assert 1 not in wg.vertices[0].adjacent
E       AssertionError

ts-3-5/test_3-5_0-9_298.py:43: AssertionError
   - [#  50] COI graph1: 
--------------------------------------------------------------------------------
  159:         if src not in self.vertices:
  160:             return
  161:         if dst not in self.vertices:
  162:             return
- 163:         if dst in self.vertices[src].adjacent:
+ 163:         if not (dst in self.vertices[src].adjacent):
  164:             self.vertices[src].remove_adjacent(dst)
--------------------------------------------------------------------------------
[0.20057 s] killed by ts-3-5/test_3-5_0-9_298.py::test_weighted_graph_remove_edge

def test_weighted_graph_remove_edge():
        wg = WeightedGraph()
        wg.add_edge(0, 1, 5)
        wg.remove_edge(0, 1)
>       assert 1 not in wg.vertices[0].adjacent
E       AssertionError

ts-3-5/test_3-5_0-9_298.py:43: AssertionError
   - [#  51] COI graph1: 
--------------------------------------------------------------------------------
  159:         if src not in self.vertices:
  160:             return
  161:         if dst not in self.vertices:
  162:             return
- 163:         if dst in self.vertices[src].adjacent:
+ 163:         if dst not in self.vertices[src].adjacent:
  164:             self.vertices[src].remove_adjacent(dst)
--------------------------------------------------------------------------------
[0.19805 s] killed by ts-3-5/test_3-5_0-9_298.py::test_weighted_graph_remove_edge

def test_weighted_graph_remove_edge():
        wg = WeightedGraph()
        wg.add_edge(0, 1, 5)
        wg.remove_edge(0, 1)
>       assert 1 not in wg.vertices[0].adjacent
E       AssertionError

ts-3-5/test_3-5_0-9_298.py:43: AssertionError
   - [#  52] OIL graph1: 
--------------------------------------------------------------------------------
   65:             visited[node] = -1
   66:             for neighbor in self.graph[node]:
   67:                 if not (valid(neighbor)):
   68:                     return False
-  69:             return True
-  70:         
+  69:                 break
+  70:             return True
   71:         for node in range(self.verticies):
   72:             
   73:             
   74:             
--------------------------------------------------------------------------------
[0.17293 s] survived
   - [#  53] OIL graph1: 
--------------------------------------------------------------------------------
   78:                     
   79:                     visited = list(map((lambda x: (abs(x))), visited))
   80:                 else:
   81:                     
-  82:                     return True
+  82:                     return Truebreak
   83:         
   84:         return False
   85:     
   86:     def topological_sort(self):
--------------------------------------------------------------------------------
[0.17972 s] survived
   - [#  54] OIL graph1: 
--------------------------------------------------------------------------------
  111:             visited.add(vertex)
  112:             for j in self.graph[vertex]:
  113:                 if j not in visited:
  114:                     dfs(j)
- 115:             
+ 115:                 break
  116:             stack.appendleft(vertex)
  117:         
  118:         for (key, _) in self.graph.items():
  119:             if key not in visited:
--------------------------------------------------------------------------------
[0.18163 s] killed by ts-3-5/test_3-5_0-9_298.py::test_graph_topological_sort

def test_graph_topological_sort():
        g = Graph(4)
        g.add_edge(0, 1)
        g.add_edge(0, 2)
        g.add_edge(1, 3)
        g.add_edge(2, 3)
>       assert g.topological_sort() == [0, 2, 1, 3]
E       AssertionError

ts-3-5/test_3-5_0-9_298.py:31: AssertionError
   - [#  55] OIL graph1: 
--------------------------------------------------------------------------------
  117:         
  118:         for (key, _) in self.graph.items():
  119:             if key not in visited:
  120:                 dfs(key)
- 121:         
+ 121:             break
  122:         return list(stack)
  123: 
  124: class WeightedGraphNode:
  125:     
--------------------------------------------------------------------------------
[0.17456 s] survived
   - [#  56] RIL graph1: 
--------------------------------------------------------------------------------
   62:                 return False
   63:             elif visited[node] == 1:
   64:                 return True
   65:             visited[node] = -1
-  66:             for neighbor in self.graph[node]:
+  66:             for neighbor in reversed(self.graph[node]):
   67:                 if not (valid(neighbor)):
   68:                     return False
   69:             return True
   70:         
--------------------------------------------------------------------------------
[0.17245 s] survived
   - [#  57] RIL graph1: 
--------------------------------------------------------------------------------
   67:                 if not (valid(neighbor)):
   68:                     return False
   69:             return True
   70:         
-  71:         for node in range(self.verticies):
+  71:         for node in reversed(range(self.verticies)):
   72:             
   73:             
   74:             
   75:             if not (visited[node]):
--------------------------------------------------------------------------------
[0.20397 s] killed by ts-3-5/test_3-5_0-9_298.py::test_graph_has_cycle

def test_graph_has_cycle():
        g = Graph(4)
        g.add_edge(0, 1)
        g.add_edge(1, 2)
        g.add_edge(2, 0)
>       assert g.has_cycle() == True

ts-3-5/test_3-5_0-9_298.py:23: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
graph1:77: in has_cycle
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

node = 3

>   ???
E   KeyError: 3

graph1:66: KeyError
   - [#  58] RIL graph1: 
--------------------------------------------------------------------------------
  108:         stack = deque()
  109:         
  110:         def dfs(vertex):
  111:             visited.add(vertex)
- 112:             for j in self.graph[vertex]:
+ 112:             for j in reversed(self.graph[vertex]):
  113:                 if j not in visited:
  114:                     dfs(j)
  115:             
  116:             stack.appendleft(vertex)
--------------------------------------------------------------------------------
[0.20028 s] killed by ts-3-5/test_3-5_0-9_298.py::test_graph_topological_sort

def test_graph_topological_sort():
        g = Graph(4)
        g.add_edge(0, 1)
        g.add_edge(0, 2)
        g.add_edge(1, 3)
        g.add_edge(2, 3)
>       assert g.topological_sort() == [0, 2, 1, 3]
E       AssertionError

ts-3-5/test_3-5_0-9_298.py:31: AssertionError
   - [#  59] RIL graph1: 
--------------------------------------------------------------------------------
  114:                     dfs(j)
  115:             
  116:             stack.appendleft(vertex)
  117:         
- 118:         for (key, _) in self.graph.items():
+ 118:         for (key, _) in reversed(self.graph.items()):
  119:             if key not in visited:
  120:                 dfs(key)
  121:         
  122:         return list(stack)
--------------------------------------------------------------------------------
[0.18517 s] killed by ts-3-5/test_3-5_0-9_298.py::test_graph_topological_sort

def test_graph_topological_sort():
        g = Graph(4)
        g.add_edge(0, 1)
        g.add_edge(0, 2)
        g.add_edge(1, 3)
        g.add_edge(2, 3)
>       assert g.topological_sort() == [0, 2, 1, 3]
E       AssertionError

ts-3-5/test_3-5_0-9_298.py:31: AssertionError
   - [#  60] ROR graph1: 
--------------------------------------------------------------------------------
   32:             source: the source vertex
   33:             destination: the destination vertex
   34: 
   35:         '''
-  36:         if len(self.graph) > self.verticies:
+  36:         if len(self.graph) < self.verticies:
   37:             raise IndexError('Too many verticies in graph.')
   38:         
   39:         if source in self.graph:
   40:             self.graph[source].append(destination)
--------------------------------------------------------------------------------
[0.17740 s] killed by ts-3-5/test_3-5_0-9_298.py::test_graph_add_edge

def test_graph_add_edge():
        g = Graph(4)
>       g.add_edge(0, 1)

ts-3-5/test_3-5_0-9_298.py:15: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph1.Graph object at 0x702648abd6f0>, source = 0, destination = 1

>   ???
E   IndexError: Too many verticies in graph.

graph1:37: IndexError
   - [#  61] ROR graph1: 
--------------------------------------------------------------------------------
   32:             source: the source vertex
   33:             destination: the destination vertex
   34: 
   35:         '''
-  36:         if len(self.graph) > self.verticies:
+  36:         if len(self.graph) >= self.verticies:
   37:             raise IndexError('Too many verticies in graph.')
   38:         
   39:         if source in self.graph:
   40:             self.graph[source].append(destination)
--------------------------------------------------------------------------------
[0.19505 s] killed by ts-3-5/test_3-5_0-9_298.py::test_graph_topological_sort

def test_graph_topological_sort():
        g = Graph(4)
        g.add_edge(0, 1)
        g.add_edge(0, 2)
        g.add_edge(1, 3)
>       g.add_edge(2, 3)

ts-3-5/test_3-5_0-9_298.py:30: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph1.Graph object at 0x702648a65360>, source = 2, destination = 3

>   ???
E   IndexError: Too many verticies in graph.

graph1:37: IndexError
   - [#  62] ROR graph1: 
--------------------------------------------------------------------------------
   57:         visited = [0] * self.verticies
   58:         
   59:         def valid(node):
   60:             
-  61:             if visited[node] == -1:
+  61:             if visited[node] != -1:
   62:                 return False
   63:             elif visited[node] == 1:
   64:                 return True
   65:             visited[node] = -1
--------------------------------------------------------------------------------
[0.19328 s] survived
   - [#  63] ROR graph1: 
--------------------------------------------------------------------------------
   59:         def valid(node):
   60:             
   61:             if visited[node] == -1:
   62:                 return False
-  63:             elif visited[node] == 1:
+  63:             elif visited[node] != 1:
   64:                 return True
   65:             visited[node] = -1
   66:             for neighbor in self.graph[node]:
   67:                 if not (valid(neighbor)):
--------------------------------------------------------------------------------
[0.19510 s] killed by ts-3-5/test_3-5_0-9_298.py::test_graph_has_cycle

def test_graph_has_cycle():
        g = Graph(4)
        g.add_edge(0, 1)
        g.add_edge(1, 2)
        g.add_edge(2, 0)
>       assert g.has_cycle() == True
E       AssertionError

ts-3-5/test_3-5_0-9_298.py:23: AssertionError
   - [#  64] SDI graph1: 
--------------------------------------------------------------------------------
    1: from collections import defaultdict, deque
    2: 
    3: class GraphNode:
    4:     
-   5:     def __init__(self, val):
-   6:         self.val = val
-   7:         self.adjacent_list = set()
-   8:     
+   5:     @staticmethod
+   6:     def __init__(self, val):
+   7:         self.val = val
+   8:         self.adjacent_list = set()
    9:     def add_adjacent(self, node):
   10:         self.adjacent_list.add(node)
   11:     
   12:     def remove_adjacent(self, node):
--------------------------------------------------------------------------------
[0.17893 s] survived
   - [#  65] SDI graph1: 
--------------------------------------------------------------------------------
    5:     def __init__(self, val):
    6:         self.val = val
    7:         self.adjacent_list = set()
    8:     
-   9:     def add_adjacent(self, node):
-  10:         self.adjacent_list.add(node)
-  11:     
+   9:     @staticmethod
+  10:     def add_adjacent(self, node):
+  11:         self.adjacent_list.add(node)
   12:     def remove_adjacent(self, node):
   13:         self.adjacent_list.remove(node)
   14: 
   15: class Graph:
--------------------------------------------------------------------------------
[0.16002 s] survived
   - [#  66] SDI graph1: 
--------------------------------------------------------------------------------
    8:     
    9:     def add_adjacent(self, node):
   10:         self.adjacent_list.add(node)
   11:     
-  12:     def remove_adjacent(self, node):
-  13:         self.adjacent_list.remove(node)
-  14: 
+  12:     @staticmethod
+  13:     def remove_adjacent(self, node):
+  14:         self.adjacent_list.remove(node)
   15: class Graph:
   16:     '''
   17:     A directed graph represented with an adjacency list.
   18:     '''
--------------------------------------------------------------------------------
[0.17274 s] survived
   - [#  67] SDI graph1: 
--------------------------------------------------------------------------------
   16:     '''
   17:     A directed graph represented with an adjacency list.
   18:     '''
   19:     
-  20:     def __init__(self, verticies):
-  21:         self.graph = {}
-  22:         self.verticies = verticies
-  23:     
+  20:     @staticmethod
+  21:     def __init__(self, verticies):
+  22:         self.graph = {}
+  23:         self.verticies = verticies
   24:     def add_edge(self, source, destination):
   25:         '''
   26:         Add an edge to the graph.
   27: 
--------------------------------------------------------------------------------
[0.19438 s] incompetent

TypeError: /home/lucca/teste/python_experiments2/graph1/ts-3-5/test_3-5_0-9_298.py:14: TypeError: Graph.__init__() missing 1 required positional argument: 'verticies'

   - [#  68] SDI graph1: 
--------------------------------------------------------------------------------
   20:     def __init__(self, verticies):
   21:         self.graph = {}
   22:         self.verticies = verticies
   23:     
-  24:     def add_edge(self, source, destination):
-  25:         '''
-  26:         Add an edge to the graph.
-  27: 
-  28:         Add an edge pointing from source vertex
-  29:         to destination vertex.
-  30: 
-  31:         Args:
-  32:             source: the source vertex
-  33:             destination: the destination vertex
-  34: 
-  35:         '''
-  36:         if len(self.graph) > self.verticies:
-  37:             raise IndexError('Too many verticies in graph.')
-  38:         
-  39:         if source in self.graph:
-  40:             self.graph[source].append(destination)
-  41:         else:
-  42:             self.graph[source] = [destination]
-  43:         
-  44:         if destination not in self.graph:
-  45:             self.graph[destination] = []
-  46:     
+  24:     @staticmethod
+  25:     def add_edge(self, source, destination):
+  26:         '''
+  27:         Add an edge to the graph.
+  28: 
+  29:         Add an edge pointing from source vertex
+  30:         to destination vertex.
+  31: 
+  32:         Args:
+  33:             source: the source vertex
+  34:             destination: the destination vertex
+  35: 
+  36:         '''
+  37:         if len(self.graph) > self.verticies:
+  38:             raise IndexError('Too many verticies in graph.')
+  39:         
+  40:         if source in self.graph:
+  41:             self.graph[source].append(destination)
+  42:         else:
+  43:             self.graph[source] = [destination]
+  44:         
+  45:         if destination not in self.graph:
+  46:             self.graph[destination] = []
   47:     def has_cycle(self):
   48:         '''
   49:         Detect if a graph has a cycle.
   50: 
--------------------------------------------------------------------------------
[0.19582 s] incompetent

TypeError: /home/lucca/teste/python_experiments2/graph1/ts-3-5/test_3-5_0-9_298.py:15: TypeError: Graph.add_edge() missing 1 required positional argument: 'destination'

   - [#  69] SDI graph1: 
--------------------------------------------------------------------------------
   43:         
   44:         if destination not in self.graph:
   45:             self.graph[destination] = []
   46:     
-  47:     def has_cycle(self):
-  48:         '''
-  49:         Detect if a graph has a cycle.
-  50: 
-  51:         Returns:
-  52:             True if the graph has a cycle and
-  53:             False if the graph is acyclic.
-  54: 
-  55:         '''
-  56:         
-  57:         visited = [0] * self.verticies
-  58:         
-  59:         def valid(node):
-  60:             
-  61:             if visited[node] == -1:
-  62:                 return False
-  63:             elif visited[node] == 1:
-  64:                 return True
-  65:             visited[node] = -1
-  66:             for neighbor in self.graph[node]:
-  67:                 if not (valid(neighbor)):
-  68:                     return False
-  69:             return True
-  70:         
-  71:         for node in range(self.verticies):
-  72:             
+  47:     @staticmethod
+  48:     def has_cycle(self):
+  49:         '''
+  50:         Detect if a graph has a cycle.
+  51: 
+  52:         Returns:
+  53:             True if the graph has a cycle and
+  54:             False if the graph is acyclic.
+  55: 
+  56:         '''
+  57:         
+  58:         visited = [0] * self.verticies
+  59:         
+  60:         def valid(node):
+  61:             
+  62:             if visited[node] == -1:
+  63:                 return False
+  64:             elif visited[node] == 1:
+  65:                 return True
+  66:             visited[node] = -1
+  67:             for neighbor in self.graph[node]:
+  68:                 if not (valid(neighbor)):
+  69:                     return False
+  70:             return True
+  71:         
+  72:         for node in range(self.verticies):
   73:             
   74:             
-  75:             if not (visited[node]):
-  76:                 
-  77:                 if valid(node):
-  78:                     
-  79:                     visited = list(map((lambda x: (abs(x))), visited))
-  80:                 else:
-  81:                     
-  82:                     return True
-  83:         
-  84:         return False
-  85:     
+  75:             
+  76:             if not (visited[node]):
+  77:                 
+  78:                 if valid(node):
+  79:                     
+  80:                     visited = list(map((lambda x: (abs(x))), visited))
+  81:                 else:
+  82:                     
+  83:                     return True
+  84:         
+  85:         return False
   86:     def topological_sort(self):
   87:         """
   88:         Sort the graph topologically.
   89: 
--------------------------------------------------------------------------------
[0.20239 s] incompetent

TypeError: /home/lucca/teste/python_experiments2/graph1/ts-3-5/test_3-5_0-9_298.py:23: TypeError: Graph.has_cycle() missing 1 required positional argument: 'self'

   - [#  70] SDI graph1: 
--------------------------------------------------------------------------------
   82:                     return True
   83:         
   84:         return False
   85:     
-  86:     def topological_sort(self):
-  87:         """
-  88:         Sort the graph topologically.
-  89: 
-  90:         A topological sort lists nodes in such a way
-  91:         that every node 's' in 's' -> 'd' directed pairs
-  92:         is listed before 'd.'  This will not work in a 
-  93:         graph that contains cycles.
-  94: 
-  95:         The algorithm looks at every node, and does a
-  96:         dfs for each node adjacent to the node and then adds
-  97:         the originating node to a stack once all adjacent
-  98:         nodes have been searched.  In the end, the stack
-  99:         will be in order of a possible topological sort.
- 100: 
- 101:         Topological sorts are not necessarily unique.
- 102: 
- 103:         Returns:
- 104:             A list of vertices in a topological ordering.
- 105: 
- 106:         """
- 107:         visited = set()
- 108:         stack = deque()
- 109:         
- 110:         def dfs(vertex):
- 111:             visited.add(vertex)
- 112:             for j in self.graph[vertex]:
- 113:                 if j not in visited:
- 114:                     dfs(j)
- 115:             
- 116:             stack.appendleft(vertex)
- 117:         
- 118:         for (key, _) in self.graph.items():
- 119:             if key not in visited:
- 120:                 dfs(key)
- 121:         
- 122:         return list(stack)
- 123: 
+  86:     @staticmethod
+  87:     def topological_sort(self):
+  88:         """
+  89:         Sort the graph topologically.
+  90: 
+  91:         A topological sort lists nodes in such a way
+  92:         that every node 's' in 's' -> 'd' directed pairs
+  93:         is listed before 'd.'  This will not work in a 
+  94:         graph that contains cycles.
+  95: 
+  96:         The algorithm looks at every node, and does a
+  97:         dfs for each node adjacent to the node and then adds
+  98:         the originating node to a stack once all adjacent
+  99:         nodes have been searched.  In the end, the stack
+ 100:         will be in order of a possible topological sort.
+ 101: 
+ 102:         Topological sorts are not necessarily unique.
+ 103: 
+ 104:         Returns:
+ 105:             A list of vertices in a topological ordering.
+ 106: 
+ 107:         """
+ 108:         visited = set()
+ 109:         stack = deque()
+ 110:         
+ 111:         def dfs(vertex):
+ 112:             visited.add(vertex)
+ 113:             for j in self.graph[vertex]:
+ 114:                 if j not in visited:
+ 115:                     dfs(j)
+ 116:             
+ 117:             stack.appendleft(vertex)
+ 118:         
+ 119:         for (key, _) in self.graph.items():
+ 120:             if key not in visited:
+ 121:                 dfs(key)
+ 122:         
+ 123:         return list(stack)
  124: class WeightedGraphNode:
  125:     
  126:     def __init__(self, val):
  127:         self.val = val
--------------------------------------------------------------------------------
[0.20392 s] incompetent

TypeError: /home/lucca/teste/python_experiments2/graph1/ts-3-5/test_3-5_0-9_298.py:31: TypeError: Graph.topological_sort() missing 1 required positional argument: 'self'

   - [#  71] SDI graph1: 
--------------------------------------------------------------------------------
  122:         return list(stack)
  123: 
  124: class WeightedGraphNode:
  125:     
- 126:     def __init__(self, val):
- 127:         self.val = val
- 128:         self.adjacent = {}
- 129:     
+ 126:     @staticmethod
+ 127:     def __init__(self, val):
+ 128:         self.val = val
+ 129:         self.adjacent = {}
  130:     def __str__(self):
  131:         return (str(self.val) + ' adjacent: ') + str([x for x in self.adjacent])
  132:     
  133:     def add_adjacent(self, node, weight=0):
--------------------------------------------------------------------------------
[0.19758 s] incompetent

TypeError: graph1:148: TypeError: WeightedGraphNode.__init__() missing 1 required positional argument: 'val'

   - [#  72] SDI graph1: 
--------------------------------------------------------------------------------
  126:     def __init__(self, val):
  127:         self.val = val
  128:         self.adjacent = {}
  129:     
- 130:     def __str__(self):
- 131:         return (str(self.val) + ' adjacent: ') + str([x for x in self.adjacent])
- 132:     
+ 130:     @staticmethod
+ 131:     def __str__(self):
+ 132:         return (str(self.val) + ' adjacent: ') + str([x for x in self.adjacent])
  133:     def add_adjacent(self, node, weight=0):
  134:         self.adjacent[node] = weight
  135:     
  136:     def remove_adjacent(self, node):
--------------------------------------------------------------------------------
[0.17714 s] survived
   - [#  73] SDI graph1: 
--------------------------------------------------------------------------------
  129:     
  130:     def __str__(self):
  131:         return (str(self.val) + ' adjacent: ') + str([x for x in self.adjacent])
  132:     
- 133:     def add_adjacent(self, node, weight=0):
- 134:         self.adjacent[node] = weight
- 135:     
+ 133:     @staticmethod
+ 134:     def add_adjacent(self, node, weight=0):
+ 135:         self.adjacent[node] = weight
  136:     def remove_adjacent(self, node):
  137:         del self.adjacent[node]
  138: 
  139: class WeightedGraph:
--------------------------------------------------------------------------------
[0.20460 s] killed by ts-3-5/test_3-5_0-9_298.py::test_weighted_graph_add_edge

def test_weighted_graph_add_edge():
        wg = WeightedGraph()
>       wg.add_edge(0, 1, 5)

ts-3-5/test_3-5_0-9_298.py:36: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
graph1:156: in add_edge
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = 1, node = 5, weight = 0

>   ???
E   AttributeError: 'int' object has no attribute 'adjacent'

graph1:135: AttributeError
   - [#  74] SDI graph1: 
--------------------------------------------------------------------------------
  132:     
  133:     def add_adjacent(self, node, weight=0):
  134:         self.adjacent[node] = weight
  135:     
- 136:     def remove_adjacent(self, node):
- 137:         del self.adjacent[node]
- 138: 
+ 136:     @staticmethod
+ 137:     def remove_adjacent(self, node):
+ 138:         del self.adjacent[node]
  139: class WeightedGraph:
  140:     
  141:     def __init__(self):
  142:         self.vertices = {}
--------------------------------------------------------------------------------
[0.19946 s] incompetent

TypeError: graph1:164: TypeError: WeightedGraphNode.remove_adjacent() missing 1 required positional argument: 'node'

   - [#  75] SDI graph1: 
--------------------------------------------------------------------------------
  137:         del self.adjacent[node]
  138: 
  139: class WeightedGraph:
  140:     
- 141:     def __init__(self):
- 142:         self.vertices = {}
- 143:     
+ 141:     @staticmethod
+ 142:     def __init__(self):
+ 143:         self.vertices = {}
  144:     def __str__(self):
  145:         return ', '.join([node.__str__() for node in self.vertices.values()])
  146:     
  147:     def add_vertex(self, val):
--------------------------------------------------------------------------------
[0.20088 s] incompetent

TypeError: /home/lucca/teste/python_experiments2/graph1/ts-3-5/test_3-5_0-9_298.py:35: TypeError: WeightedGraph.__init__() missing 1 required positional argument: 'self'

   - [#  76] SDI graph1: 
--------------------------------------------------------------------------------
  140:     
  141:     def __init__(self):
  142:         self.vertices = {}
  143:     
- 144:     def __str__(self):
- 145:         return ', '.join([node.__str__() for node in self.vertices.values()])
- 146:     
+ 144:     @staticmethod
+ 145:     def __str__(self):
+ 146:         return ', '.join([node.__str__() for node in self.vertices.values()])
  147:     def add_vertex(self, val):
  148:         vertex = WeightedGraphNode(val)
  149:         self.vertices[val] = vertex
  150:     
--------------------------------------------------------------------------------
[0.18149 s] survived
   - [#  77] SDI graph1: 
--------------------------------------------------------------------------------
  143:     
  144:     def __str__(self):
  145:         return ', '.join([node.__str__() for node in self.vertices.values()])
  146:     
- 147:     def add_vertex(self, val):
- 148:         vertex = WeightedGraphNode(val)
- 149:         self.vertices[val] = vertex
- 150:     
+ 147:     @staticmethod
+ 148:     def add_vertex(self, val):
+ 149:         vertex = WeightedGraphNode(val)
+ 150:         self.vertices[val] = vertex
  151:     def add_edge(self, src, dst, weight=0):
  152:         if src not in self.vertices:
  153:             self.add_vertex(src)
  154:         if dst not in self.vertices:
--------------------------------------------------------------------------------
[0.19710 s] incompetent

TypeError: graph1:153: TypeError: WeightedGraph.add_vertex() missing 1 required positional argument: 'val'

   - [#  78] SDI graph1: 
--------------------------------------------------------------------------------
  147:     def add_vertex(self, val):
  148:         vertex = WeightedGraphNode(val)
  149:         self.vertices[val] = vertex
  150:     
- 151:     def add_edge(self, src, dst, weight=0):
- 152:         if src not in self.vertices:
- 153:             self.add_vertex(src)
- 154:         if dst not in self.vertices:
- 155:             self.add_vertex(dst)
- 156:         self.vertices[src].add_adjacent(dst, weight)
- 157:     
+ 151:     @staticmethod
+ 152:     def add_edge(self, src, dst, weight=0):
+ 153:         if src not in self.vertices:
+ 154:             self.add_vertex(src)
+ 155:         if dst not in self.vertices:
+ 156:             self.add_vertex(dst)
+ 157:         self.vertices[src].add_adjacent(dst, weight)
  158:     def remove_edge(self, src, dst):
  159:         if src not in self.vertices:
  160:             return
  161:         if dst not in self.vertices:
--------------------------------------------------------------------------------
[0.20197 s] killed by ts-3-5/test_3-5_0-9_298.py::test_weighted_graph_add_edge

def test_weighted_graph_add_edge():
        wg = WeightedGraph()
>       wg.add_edge(0, 1, 5)

ts-3-5/test_3-5_0-9_298.py:36: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = 0, src = 1, dst = 5, weight = 0

>   ???
E   AttributeError: 'int' object has no attribute 'vertices'

graph1:153: AttributeError
   - [#  79] SDI graph1: 
--------------------------------------------------------------------------------
  154:         if dst not in self.vertices:
  155:             self.add_vertex(dst)
  156:         self.vertices[src].add_adjacent(dst, weight)
  157:     
- 158:     def remove_edge(self, src, dst):
- 159:         if src not in self.vertices:
- 160:             return
- 161:         if dst not in self.vertices:
- 162:             return
- 163:         if dst in self.vertices[src].adjacent:
- 164:             self.vertices[src].remove_adjacent(dst)
+ 158:     @staticmethod
+ 159:     def remove_edge(self, src, dst):
+ 160:         if src not in self.vertices:
+ 161:             return
+ 162:         if dst not in self.vertices:
+ 163:             return
+ 164:         if dst in self.vertices[src].adjacent:
+ 165:             self.vertices[src].remove_adjacent(dst)
--------------------------------------------------------------------------------
[0.20018 s] incompetent

TypeError: /home/lucca/teste/python_experiments2/graph1/ts-3-5/test_3-5_0-9_298.py:42: TypeError: WeightedGraph.remove_edge() missing 1 required positional argument: 'dst'

   - [#  80] SDL graph1: 
--------------------------------------------------------------------------------
    2: 
    3: class GraphNode:
    4:     
    5:     def __init__(self, val):
-   6:         self.val = val
+   6:         pass
    7:         self.adjacent_list = set()
    8:     
    9:     def add_adjacent(self, node):
   10:         self.adjacent_list.add(node)
--------------------------------------------------------------------------------
[0.17584 s] survived
   - [#  81] SDL graph1: 
--------------------------------------------------------------------------------
    3: class GraphNode:
    4:     
    5:     def __init__(self, val):
    6:         self.val = val
-   7:         self.adjacent_list = set()
+   7:         pass
    8:     
    9:     def add_adjacent(self, node):
   10:         self.adjacent_list.add(node)
   11:     
--------------------------------------------------------------------------------
[0.19740 s] survived
   - [#  82] SDL graph1: 
--------------------------------------------------------------------------------
    6:         self.val = val
    7:         self.adjacent_list = set()
    8:     
    9:     def add_adjacent(self, node):
-  10:         self.adjacent_list.add(node)
+  10:         pass
   11:     
   12:     def remove_adjacent(self, node):
   13:         self.adjacent_list.remove(node)
   14: 
--------------------------------------------------------------------------------
[0.16440 s] survived
   - [#  83] SDL graph1: 
--------------------------------------------------------------------------------
    9:     def add_adjacent(self, node):
   10:         self.adjacent_list.add(node)
   11:     
   12:     def remove_adjacent(self, node):
-  13:         self.adjacent_list.remove(node)
+  13:         pass
   14: 
   15: class Graph:
   16:     '''
   17:     A directed graph represented with an adjacency list.
--------------------------------------------------------------------------------
[0.15616 s] survived
   - [#  84] SDL graph1: 
--------------------------------------------------------------------------------
   17:     A directed graph represented with an adjacency list.
   18:     '''
   19:     
   20:     def __init__(self, verticies):
-  21:         self.graph = {}
+  21:         pass
   22:         self.verticies = verticies
   23:     
   24:     def add_edge(self, source, destination):
   25:         '''
--------------------------------------------------------------------------------
[0.19546 s] killed by ts-3-5/test_3-5_0-9_298.py::test_graph_add_edge

def test_graph_add_edge():
        g = Graph(4)
>       g.add_edge(0, 1)

ts-3-5/test_3-5_0-9_298.py:15: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph1.Graph object at 0x70264880db40>, source = 0, destination = 1

>   ???
E   AttributeError: 'Graph' object has no attribute 'graph'

graph1:36: AttributeError
   - [#  85] SDL graph1: 
--------------------------------------------------------------------------------
   18:     '''
   19:     
   20:     def __init__(self, verticies):
   21:         self.graph = {}
-  22:         self.verticies = verticies
+  22:         pass
   23:     
   24:     def add_edge(self, source, destination):
   25:         '''
   26:         Add an edge to the graph.
--------------------------------------------------------------------------------
[0.19745 s] killed by ts-3-5/test_3-5_0-9_298.py::test_graph_add_edge

def test_graph_add_edge():
        g = Graph(4)
>       g.add_edge(0, 1)

ts-3-5/test_3-5_0-9_298.py:15: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph1.Graph object at 0x7026488aefe0>, source = 0, destination = 1

>   ???
E   AttributeError: 'Graph' object has no attribute 'verticies'

graph1:36: AttributeError
   - [#  86] SDL graph1: 
--------------------------------------------------------------------------------
   36:         if len(self.graph) > self.verticies:
   37:             raise IndexError('Too many verticies in graph.')
   38:         
   39:         if source in self.graph:
-  40:             self.graph[source].append(destination)
+  40:             pass
   41:         else:
   42:             self.graph[source] = [destination]
   43:         
   44:         if destination not in self.graph:
--------------------------------------------------------------------------------
[0.20101 s] killed by ts-3-5/test_3-5_0-9_298.py::test_graph_has_cycle

def test_graph_has_cycle():
        g = Graph(4)
        g.add_edge(0, 1)
        g.add_edge(1, 2)
        g.add_edge(2, 0)
>       assert g.has_cycle() == True

ts-3-5/test_3-5_0-9_298.py:23: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
graph1:77: in has_cycle
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

node = 3

>   ???
E   KeyError: 3

graph1:66: KeyError
   - [#  87] SDL graph1: 
--------------------------------------------------------------------------------
   38:         
   39:         if source in self.graph:
   40:             self.graph[source].append(destination)
   41:         else:
-  42:             self.graph[source] = [destination]
+  42:             pass
   43:         
   44:         if destination not in self.graph:
   45:             self.graph[destination] = []
   46:     
--------------------------------------------------------------------------------
[0.20064 s] killed by ts-3-5/test_3-5_0-9_298.py::test_graph_add_edge

def test_graph_add_edge():
        g = Graph(4)
        g.add_edge(0, 1)
>       assert g.graph == {0: [1], 1: []}
E       AssertionError

ts-3-5/test_3-5_0-9_298.py:16: AssertionError
   - [#  88] SDL graph1: 
--------------------------------------------------------------------------------
   41:         else:
   42:             self.graph[source] = [destination]
   43:         
   44:         if destination not in self.graph:
-  45:             self.graph[destination] = []
+  45:             pass
   46:     
   47:     def has_cycle(self):
   48:         '''
   49:         Detect if a graph has a cycle.
--------------------------------------------------------------------------------
[0.19904 s] killed by ts-3-5/test_3-5_0-9_298.py::test_graph_add_edge

def test_graph_add_edge():
        g = Graph(4)
        g.add_edge(0, 1)
>       assert g.graph == {0: [1], 1: []}
E       AssertionError

ts-3-5/test_3-5_0-9_298.py:16: AssertionError
   - [#  89] SDL graph1: 
--------------------------------------------------------------------------------
   53:             False if the graph is acyclic.
   54: 
   55:         '''
   56:         
-  57:         visited = [0] * self.verticies
+  57:         pass
   58:         
   59:         def valid(node):
   60:             
   61:             if visited[node] == -1:
--------------------------------------------------------------------------------
[0.19727 s] killed by ts-3-5/test_3-5_0-9_298.py::test_graph_has_cycle

def test_graph_has_cycle():
        g = Graph(4)
        g.add_edge(0, 1)
        g.add_edge(1, 2)
        g.add_edge(2, 0)
>       assert g.has_cycle() == True

ts-3-5/test_3-5_0-9_298.py:23: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph1.Graph object at 0x702648bf6ce0>

>   ???
E   UnboundLocalError: local variable 'visited' referenced before assignment

graph1:75: UnboundLocalError
   - [#  90] SDL graph1: 
--------------------------------------------------------------------------------
   58:         
   59:         def valid(node):
   60:             
   61:             if visited[node] == -1:
-  62:                 return False
+  62:                 pass
   63:             elif visited[node] == 1:
   64:                 return True
   65:             visited[node] = -1
   66:             for neighbor in self.graph[node]:
--------------------------------------------------------------------------------
[0.20522 s] killed by ts-3-5/test_3-5_0-9_298.py::test_graph_has_cycle

def test_graph_has_cycle():
        g = Graph(4)
        g.add_edge(0, 1)
        g.add_edge(1, 2)
        g.add_edge(2, 0)
>       assert g.has_cycle() == True

ts-3-5/test_3-5_0-9_298.py:23: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
graph1:77: in has_cycle
    ???
graph1:67: in valid
    ???
graph1:67: in valid
    ???
graph1:67: in valid
    ???
graph1:67: in valid
    ???
E   RecursionError: maximum recursion depth exceeded in comparison
!!! Recursion detected (same locals & position)
   - [#  91] SDL graph1: 
--------------------------------------------------------------------------------
   60:             
   61:             if visited[node] == -1:
   62:                 return False
   63:             elif visited[node] == 1:
-  64:                 return True
+  64:                 pass
   65:             visited[node] = -1
   66:             for neighbor in self.graph[node]:
   67:                 if not (valid(neighbor)):
   68:                     return False
--------------------------------------------------------------------------------
[0.17632 s] survived
   - [#  92] SDL graph1: 
--------------------------------------------------------------------------------
   61:             if visited[node] == -1:
   62:                 return False
   63:             elif visited[node] == 1:
   64:                 return True
-  65:             visited[node] = -1
+  65:             pass
   66:             for neighbor in self.graph[node]:
   67:                 if not (valid(neighbor)):
   68:                     return False
   69:             return True
--------------------------------------------------------------------------------
[0.20192 s] killed by ts-3-5/test_3-5_0-9_298.py::test_graph_has_cycle

def test_graph_has_cycle():
        g = Graph(4)
        g.add_edge(0, 1)
        g.add_edge(1, 2)
        g.add_edge(2, 0)
>       assert g.has_cycle() == True

ts-3-5/test_3-5_0-9_298.py:23: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
graph1:77: in has_cycle
    ???
graph1:67: in valid
    ???
graph1:67: in valid
    ???
graph1:67: in valid
    ???
graph1:67: in valid
    ???
E   RecursionError: maximum recursion depth exceeded in comparison
!!! Recursion detected (same locals & position)
   - [#  93] SDL graph1: 
--------------------------------------------------------------------------------
   64:                 return True
   65:             visited[node] = -1
   66:             for neighbor in self.graph[node]:
   67:                 if not (valid(neighbor)):
-  68:                     return False
+  68:                     pass
   69:             return True
   70:         
   71:         for node in range(self.verticies):
   72:             
--------------------------------------------------------------------------------
[0.20106 s] killed by ts-3-5/test_3-5_0-9_298.py::test_graph_has_cycle

def test_graph_has_cycle():
        g = Graph(4)
        g.add_edge(0, 1)
        g.add_edge(1, 2)
        g.add_edge(2, 0)
>       assert g.has_cycle() == True

ts-3-5/test_3-5_0-9_298.py:23: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
graph1:77: in has_cycle
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

node = 3

>   ???
E   KeyError: 3

graph1:66: KeyError
   - [#  94] SDL graph1: 
--------------------------------------------------------------------------------
   65:             visited[node] = -1
   66:             for neighbor in self.graph[node]:
   67:                 if not (valid(neighbor)):
   68:                     return False
-  69:             return True
+  69:             pass
   70:         
   71:         for node in range(self.verticies):
   72:             
   73:             
--------------------------------------------------------------------------------
[0.17649 s] survived
   - [#  95] SDL graph1: 
--------------------------------------------------------------------------------
   75:             if not (visited[node]):
   76:                 
   77:                 if valid(node):
   78:                     
-  79:                     visited = list(map((lambda x: (abs(x))), visited))
+  79:                     pass
   80:                 else:
   81:                     
   82:                     return True
   83:         
--------------------------------------------------------------------------------
[0.17828 s] survived
   - [#  96] SDL graph1: 
--------------------------------------------------------------------------------
   78:                     
   79:                     visited = list(map((lambda x: (abs(x))), visited))
   80:                 else:
   81:                     
-  82:                     return True
+  82:                     pass
   83:         
   84:         return False
   85:     
   86:     def topological_sort(self):
--------------------------------------------------------------------------------
[0.20097 s] killed by ts-3-5/test_3-5_0-9_298.py::test_graph_has_cycle

def test_graph_has_cycle():
        g = Graph(4)
        g.add_edge(0, 1)
        g.add_edge(1, 2)
        g.add_edge(2, 0)
>       assert g.has_cycle() == True

ts-3-5/test_3-5_0-9_298.py:23: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
graph1:77: in has_cycle
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

node = 3

>   ???
E   KeyError: 3

graph1:66: KeyError
   - [#  97] SDL graph1: 
--------------------------------------------------------------------------------
   80:                 else:
   81:                     
   82:                     return True
   83:         
-  84:         return False
+  84:         pass
   85:     
   86:     def topological_sort(self):
   87:         """
   88:         Sort the graph topologically.
--------------------------------------------------------------------------------
[0.17711 s] survived
   - [#  98] SDL graph1: 
--------------------------------------------------------------------------------
  103:         Returns:
  104:             A list of vertices in a topological ordering.
  105: 
  106:         """
- 107:         visited = set()
+ 107:         pass
  108:         stack = deque()
  109:         
  110:         def dfs(vertex):
  111:             visited.add(vertex)
--------------------------------------------------------------------------------
[0.20103 s] killed by ts-3-5/test_3-5_0-9_298.py::test_graph_topological_sort

def test_graph_topological_sort():
        g = Graph(4)
        g.add_edge(0, 1)
        g.add_edge(0, 2)
        g.add_edge(1, 3)
        g.add_edge(2, 3)
>       assert g.topological_sort() == [0, 2, 1, 3]

ts-3-5/test_3-5_0-9_298.py:31: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph1.Graph object at 0x702648ae1a80>

>   ???
E   NameError: name 'visited' is not defined

graph1:119: NameError
   - [#  99] SDL graph1: 
--------------------------------------------------------------------------------
  104:             A list of vertices in a topological ordering.
  105: 
  106:         """
  107:         visited = set()
- 108:         stack = deque()
+ 108:         pass
  109:         
  110:         def dfs(vertex):
  111:             visited.add(vertex)
  112:             for j in self.graph[vertex]:
--------------------------------------------------------------------------------
[0.22407 s] killed by ts-3-5/test_3-5_0-9_298.py::test_graph_topological_sort

def test_graph_topological_sort():
        g = Graph(4)
        g.add_edge(0, 1)
        g.add_edge(0, 2)
        g.add_edge(1, 3)
        g.add_edge(2, 3)
>       assert g.topological_sort() == [0, 2, 1, 3]

ts-3-5/test_3-5_0-9_298.py:31: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
graph1:120: in topological_sort
    ???
graph1:114: in dfs
    ???
graph1:114: in dfs
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

vertex = 3

>   ???
E   NameError: name 'stack' is not defined

graph1:116: NameError
   - [# 100] SDL graph1: 
--------------------------------------------------------------------------------
  107:         visited = set()
  108:         stack = deque()
  109:         
  110:         def dfs(vertex):
- 111:             visited.add(vertex)
+ 111:             pass
  112:             for j in self.graph[vertex]:
  113:                 if j not in visited:
  114:                     dfs(j)
  115:             
--------------------------------------------------------------------------------
[0.19741 s] killed by ts-3-5/test_3-5_0-9_298.py::test_graph_topological_sort

def test_graph_topological_sort():
        g = Graph(4)
        g.add_edge(0, 1)
        g.add_edge(0, 2)
        g.add_edge(1, 3)
        g.add_edge(2, 3)
>       assert g.topological_sort() == [0, 2, 1, 3]
E       AssertionError

ts-3-5/test_3-5_0-9_298.py:31: AssertionError
   - [# 101] SDL graph1: 
--------------------------------------------------------------------------------
  110:         def dfs(vertex):
  111:             visited.add(vertex)
  112:             for j in self.graph[vertex]:
  113:                 if j not in visited:
- 114:                     dfs(j)
+ 114:                     pass
  115:             
  116:             stack.appendleft(vertex)
  117:         
  118:         for (key, _) in self.graph.items():
--------------------------------------------------------------------------------
[0.19799 s] killed by ts-3-5/test_3-5_0-9_298.py::test_graph_topological_sort

def test_graph_topological_sort():
        g = Graph(4)
        g.add_edge(0, 1)
        g.add_edge(0, 2)
        g.add_edge(1, 3)
        g.add_edge(2, 3)
>       assert g.topological_sort() == [0, 2, 1, 3]
E       AssertionError

ts-3-5/test_3-5_0-9_298.py:31: AssertionError
   - [# 102] SDL graph1: 
--------------------------------------------------------------------------------
  112:             for j in self.graph[vertex]:
  113:                 if j not in visited:
  114:                     dfs(j)
  115:             
- 116:             stack.appendleft(vertex)
+ 116:             pass
  117:         
  118:         for (key, _) in self.graph.items():
  119:             if key not in visited:
  120:                 dfs(key)
--------------------------------------------------------------------------------
[0.19898 s] killed by ts-3-5/test_3-5_0-9_298.py::test_graph_topological_sort

def test_graph_topological_sort():
        g = Graph(4)
        g.add_edge(0, 1)
        g.add_edge(0, 2)
        g.add_edge(1, 3)
        g.add_edge(2, 3)
>       assert g.topological_sort() == [0, 2, 1, 3]
E       AssertionError

ts-3-5/test_3-5_0-9_298.py:31: AssertionError
   - [# 103] SDL graph1: 
--------------------------------------------------------------------------------
  116:             stack.appendleft(vertex)
  117:         
  118:         for (key, _) in self.graph.items():
  119:             if key not in visited:
- 120:                 dfs(key)
+ 120:                 pass
  121:         
  122:         return list(stack)
  123: 
  124: class WeightedGraphNode:
--------------------------------------------------------------------------------
[0.20052 s] killed by ts-3-5/test_3-5_0-9_298.py::test_graph_topological_sort

def test_graph_topological_sort():
        g = Graph(4)
        g.add_edge(0, 1)
        g.add_edge(0, 2)
        g.add_edge(1, 3)
        g.add_edge(2, 3)
>       assert g.topological_sort() == [0, 2, 1, 3]
E       AssertionError

ts-3-5/test_3-5_0-9_298.py:31: AssertionError
   - [# 104] SDL graph1: 
--------------------------------------------------------------------------------
  118:         for (key, _) in self.graph.items():
  119:             if key not in visited:
  120:                 dfs(key)
  121:         
- 122:         return list(stack)
+ 122:         pass
  123: 
  124: class WeightedGraphNode:
  125:     
  126:     def __init__(self, val):
--------------------------------------------------------------------------------
[0.20299 s] killed by ts-3-5/test_3-5_0-9_298.py::test_graph_topological_sort

def test_graph_topological_sort():
        g = Graph(4)
        g.add_edge(0, 1)
        g.add_edge(0, 2)
        g.add_edge(1, 3)
        g.add_edge(2, 3)
>       assert g.topological_sort() == [0, 2, 1, 3]
E       AssertionError

ts-3-5/test_3-5_0-9_298.py:31: AssertionError
   - [# 105] SDL graph1: 
--------------------------------------------------------------------------------
  123: 
  124: class WeightedGraphNode:
  125:     
  126:     def __init__(self, val):
- 127:         self.val = val
+ 127:         pass
  128:         self.adjacent = {}
  129:     
  130:     def __str__(self):
  131:         return (str(self.val) + ' adjacent: ') + str([x for x in self.adjacent])
--------------------------------------------------------------------------------
[0.17665 s] survived
   - [# 106] SDL graph1: 
--------------------------------------------------------------------------------
  124: class WeightedGraphNode:
  125:     
  126:     def __init__(self, val):
  127:         self.val = val
- 128:         self.adjacent = {}
+ 128:         pass
  129:     
  130:     def __str__(self):
  131:         return (str(self.val) + ' adjacent: ') + str([x for x in self.adjacent])
  132:     
--------------------------------------------------------------------------------
[0.19590 s] killed by ts-3-5/test_3-5_0-9_298.py::test_weighted_graph_add_edge

def test_weighted_graph_add_edge():
        wg = WeightedGraph()
>       wg.add_edge(0, 1, 5)

ts-3-5/test_3-5_0-9_298.py:36: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
graph1:156: in add_edge
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph1.WeightedGraphNode object at 0x702648ad9d20>, node = 1, weight = 5

>   ???
E   AttributeError: 'WeightedGraphNode' object has no attribute 'adjacent'

graph1:134: AttributeError
   - [# 107] SDL graph1: 
--------------------------------------------------------------------------------
  127:         self.val = val
  128:         self.adjacent = {}
  129:     
  130:     def __str__(self):
- 131:         return (str(self.val) + ' adjacent: ') + str([x for x in self.adjacent])
+ 131:         pass
  132:     
  133:     def add_adjacent(self, node, weight=0):
  134:         self.adjacent[node] = weight
  135:     
--------------------------------------------------------------------------------
[0.17406 s] survived
   - [# 108] SDL graph1: 
--------------------------------------------------------------------------------
  130:     def __str__(self):
  131:         return (str(self.val) + ' adjacent: ') + str([x for x in self.adjacent])
  132:     
  133:     def add_adjacent(self, node, weight=0):
- 134:         self.adjacent[node] = weight
+ 134:         pass
  135:     
  136:     def remove_adjacent(self, node):
  137:         del self.adjacent[node]
  138: 
--------------------------------------------------------------------------------
[0.20120 s] killed by ts-3-5/test_3-5_0-9_298.py::test_weighted_graph_add_edge

def test_weighted_graph_add_edge():
        wg = WeightedGraph()
        wg.add_edge(0, 1, 5)
>       assert wg.vertices[0].adjacent == {1: 5}
E       AssertionError

ts-3-5/test_3-5_0-9_298.py:37: AssertionError
   - [# 109] SDL graph1: 
--------------------------------------------------------------------------------
  138: 
  139: class WeightedGraph:
  140:     
  141:     def __init__(self):
- 142:         self.vertices = {}
+ 142:         pass
  143:     
  144:     def __str__(self):
  145:         return ', '.join([node.__str__() for node in self.vertices.values()])
  146:     
--------------------------------------------------------------------------------
[0.19983 s] killed by ts-3-5/test_3-5_0-9_298.py::test_weighted_graph_add_edge

def test_weighted_graph_add_edge():
        wg = WeightedGraph()
>       wg.add_edge(0, 1, 5)

ts-3-5/test_3-5_0-9_298.py:36: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph1.WeightedGraph object at 0x7026488af6d0>, src = 0, dst = 1
weight = 5

>   ???
E   AttributeError: 'WeightedGraph' object has no attribute 'vertices'

graph1:152: AttributeError
   - [# 110] SDL graph1: 
--------------------------------------------------------------------------------
  141:     def __init__(self):
  142:         self.vertices = {}
  143:     
  144:     def __str__(self):
- 145:         return ', '.join([node.__str__() for node in self.vertices.values()])
+ 145:         pass
  146:     
  147:     def add_vertex(self, val):
  148:         vertex = WeightedGraphNode(val)
  149:         self.vertices[val] = vertex
--------------------------------------------------------------------------------
[0.17834 s] survived
   - [# 111] SDL graph1: 
--------------------------------------------------------------------------------
  144:     def __str__(self):
  145:         return ', '.join([node.__str__() for node in self.vertices.values()])
  146:     
  147:     def add_vertex(self, val):
- 148:         vertex = WeightedGraphNode(val)
+ 148:         pass
  149:         self.vertices[val] = vertex
  150:     
  151:     def add_edge(self, src, dst, weight=0):
  152:         if src not in self.vertices:
--------------------------------------------------------------------------------
[0.18021 s] killed by ts-3-5/test_3-5_0-9_298.py::test_weighted_graph_add_edge

def test_weighted_graph_add_edge():
        wg = WeightedGraph()
>       wg.add_edge(0, 1, 5)

ts-3-5/test_3-5_0-9_298.py:36: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
graph1:153: in add_edge
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph1.WeightedGraph object at 0x70264876dae0>, val = 0

>   ???
E   NameError: name 'vertex' is not defined

graph1:149: NameError
   - [# 112] SDL graph1: 
--------------------------------------------------------------------------------
  145:         return ', '.join([node.__str__() for node in self.vertices.values()])
  146:     
  147:     def add_vertex(self, val):
  148:         vertex = WeightedGraphNode(val)
- 149:         self.vertices[val] = vertex
+ 149:         pass
  150:     
  151:     def add_edge(self, src, dst, weight=0):
  152:         if src not in self.vertices:
  153:             self.add_vertex(src)
--------------------------------------------------------------------------------
[0.19621 s] killed by ts-3-5/test_3-5_0-9_298.py::test_weighted_graph_add_edge

def test_weighted_graph_add_edge():
        wg = WeightedGraph()
>       wg.add_edge(0, 1, 5)

ts-3-5/test_3-5_0-9_298.py:36: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph1.WeightedGraph object at 0x702648a32230>, src = 0, dst = 1
weight = 5

>   ???
E   KeyError: 0

graph1:156: KeyError
   - [# 113] SDL graph1: 
--------------------------------------------------------------------------------
  149:         self.vertices[val] = vertex
  150:     
  151:     def add_edge(self, src, dst, weight=0):
  152:         if src not in self.vertices:
- 153:             self.add_vertex(src)
+ 153:             pass
  154:         if dst not in self.vertices:
  155:             self.add_vertex(dst)
  156:         self.vertices[src].add_adjacent(dst, weight)
  157:     
--------------------------------------------------------------------------------
[0.19947 s] killed by ts-3-5/test_3-5_0-9_298.py::test_weighted_graph_add_edge

def test_weighted_graph_add_edge():
        wg = WeightedGraph()
>       wg.add_edge(0, 1, 5)

ts-3-5/test_3-5_0-9_298.py:36: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph1.WeightedGraph object at 0x70264876d000>, src = 0, dst = 1
weight = 5

>   ???
E   KeyError: 0

graph1:156: KeyError
   - [# 114] SDL graph1: 
--------------------------------------------------------------------------------
  151:     def add_edge(self, src, dst, weight=0):
  152:         if src not in self.vertices:
  153:             self.add_vertex(src)
  154:         if dst not in self.vertices:
- 155:             self.add_vertex(dst)
+ 155:             pass
  156:         self.vertices[src].add_adjacent(dst, weight)
  157:     
  158:     def remove_edge(self, src, dst):
  159:         if src not in self.vertices:
--------------------------------------------------------------------------------
[0.20167 s] killed by ts-3-5/test_3-5_0-9_298.py::test_weighted_graph_remove_edge

def test_weighted_graph_remove_edge():
        wg = WeightedGraph()
        wg.add_edge(0, 1, 5)
        wg.remove_edge(0, 1)
>       assert 1 not in wg.vertices[0].adjacent
E       AssertionError

ts-3-5/test_3-5_0-9_298.py:43: AssertionError
   - [# 115] SDL graph1: 
--------------------------------------------------------------------------------
  152:         if src not in self.vertices:
  153:             self.add_vertex(src)
  154:         if dst not in self.vertices:
  155:             self.add_vertex(dst)
- 156:         self.vertices[src].add_adjacent(dst, weight)
+ 156:         pass
  157:     
  158:     def remove_edge(self, src, dst):
  159:         if src not in self.vertices:
  160:             return
--------------------------------------------------------------------------------
[0.23512 s] killed by ts-3-5/test_3-5_0-9_298.py::test_weighted_graph_add_edge

def test_weighted_graph_add_edge():
        wg = WeightedGraph()
        wg.add_edge(0, 1, 5)
>       assert wg.vertices[0].adjacent == {1: 5}
E       AssertionError

ts-3-5/test_3-5_0-9_298.py:37: AssertionError
   - [# 116] SDL graph1: 
--------------------------------------------------------------------------------
  156:         self.vertices[src].add_adjacent(dst, weight)
  157:     
  158:     def remove_edge(self, src, dst):
  159:         if src not in self.vertices:
- 160:             return
+ 160:             pass
  161:         if dst not in self.vertices:
  162:             return
  163:         if dst in self.vertices[src].adjacent:
  164:             self.vertices[src].remove_adjacent(dst)
--------------------------------------------------------------------------------
[0.17641 s] survived
   - [# 117] SDL graph1: 
--------------------------------------------------------------------------------
  158:     def remove_edge(self, src, dst):
  159:         if src not in self.vertices:
  160:             return
  161:         if dst not in self.vertices:
- 162:             return
+ 162:             pass
  163:         if dst in self.vertices[src].adjacent:
  164:             self.vertices[src].remove_adjacent(dst)
--------------------------------------------------------------------------------
[0.17283 s] survived
   - [# 118] SDL graph1: 
--------------------------------------------------------------------------------
  160:             return
  161:         if dst not in self.vertices:
  162:             return
  163:         if dst in self.vertices[src].adjacent:
- 164:             self.vertices[src].remove_adjacent(dst)
+ 164:             pass
--------------------------------------------------------------------------------
[0.19912 s] killed by ts-3-5/test_3-5_0-9_298.py::test_weighted_graph_remove_edge

def test_weighted_graph_remove_edge():
        wg = WeightedGraph()
        wg.add_edge(0, 1, 5)
        wg.remove_edge(0, 1)
>       assert 1 not in wg.vertices[0].adjacent
E       AssertionError

ts-3-5/test_3-5_0-9_298.py:43: AssertionError
   - [# 119] SVD graph1: 
--------------------------------------------------------------------------------
    2: 
    3: class GraphNode:
    4:     
    5:     def __init__(self, val):
-   6:         self.val = val
+   6:         val = val
    7:         self.adjacent_list = set()
    8:     
    9:     def add_adjacent(self, node):
   10:         self.adjacent_list.add(node)
--------------------------------------------------------------------------------
[0.00000 s] incompetent

ValueError: expression must have Store context but has Load instead

   - [# 120] SVD graph1: 
--------------------------------------------------------------------------------
    3: class GraphNode:
    4:     
    5:     def __init__(self, val):
    6:         self.val = val
-   7:         self.adjacent_list = set()
+   7:         adjacent_list = set()
    8:     
    9:     def add_adjacent(self, node):
   10:         self.adjacent_list.add(node)
   11:     
--------------------------------------------------------------------------------
[0.00000 s] incompetent

ValueError: expression must have Store context but has Load instead

   - [# 121] SVD graph1: 
--------------------------------------------------------------------------------
    6:         self.val = val
    7:         self.adjacent_list = set()
    8:     
    9:     def add_adjacent(self, node):
-  10:         self.adjacent_list.add(node)
+  10:         adjacent_list.add(node)
   11:     
   12:     def remove_adjacent(self, node):
   13:         self.adjacent_list.remove(node)
   14: 
--------------------------------------------------------------------------------
[0.17329 s] survived
   - [# 122] SVD graph1: 
--------------------------------------------------------------------------------
    9:     def add_adjacent(self, node):
   10:         self.adjacent_list.add(node)
   11:     
   12:     def remove_adjacent(self, node):
-  13:         self.adjacent_list.remove(node)
+  13:         adjacent_list.remove(node)
   14: 
   15: class Graph:
   16:     '''
   17:     A directed graph represented with an adjacency list.
--------------------------------------------------------------------------------
[0.19876 s] survived
   - [# 123] SVD graph1: 
--------------------------------------------------------------------------------
   17:     A directed graph represented with an adjacency list.
   18:     '''
   19:     
   20:     def __init__(self, verticies):
-  21:         self.graph = {}
+  21:         graph = {}
   22:         self.verticies = verticies
   23:     
   24:     def add_edge(self, source, destination):
   25:         '''
--------------------------------------------------------------------------------
[0.00000 s] incompetent

ValueError: expression must have Store context but has Load instead

   - [# 124] SVD graph1: 
--------------------------------------------------------------------------------
   18:     '''
   19:     
   20:     def __init__(self, verticies):
   21:         self.graph = {}
-  22:         self.verticies = verticies
+  22:         verticies = verticies
   23:     
   24:     def add_edge(self, source, destination):
   25:         '''
   26:         Add an edge to the graph.
--------------------------------------------------------------------------------
[0.00000 s] incompetent

ValueError: expression must have Store context but has Load instead

   - [# 125] SVD graph1: 
--------------------------------------------------------------------------------
   32:             source: the source vertex
   33:             destination: the destination vertex
   34: 
   35:         '''
-  36:         if len(self.graph) > self.verticies:
+  36:         if len(graph) > self.verticies:
   37:             raise IndexError('Too many verticies in graph.')
   38:         
   39:         if source in self.graph:
   40:             self.graph[source].append(destination)
--------------------------------------------------------------------------------
[0.18734 s] killed by ts-3-5/test_3-5_0-9_298.py::test_graph_add_edge

def test_graph_add_edge():
        g = Graph(4)
>       g.add_edge(0, 1)

ts-3-5/test_3-5_0-9_298.py:15: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph1.Graph object at 0x702648a21510>, source = 0, destination = 1

>   ???
E   NameError: name 'graph' is not defined

graph1:36: NameError
   - [# 126] SVD graph1: 
--------------------------------------------------------------------------------
   32:             source: the source vertex
   33:             destination: the destination vertex
   34: 
   35:         '''
-  36:         if len(self.graph) > self.verticies:
+  36:         if len(self.graph) > verticies:
   37:             raise IndexError('Too many verticies in graph.')
   38:         
   39:         if source in self.graph:
   40:             self.graph[source].append(destination)
--------------------------------------------------------------------------------
[0.19493 s] killed by ts-3-5/test_3-5_0-9_298.py::test_graph_add_edge

def test_graph_add_edge():
        g = Graph(4)
>       g.add_edge(0, 1)

ts-3-5/test_3-5_0-9_298.py:15: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph1.Graph object at 0x702648ff06a0>, source = 0, destination = 1

>   ???
E   NameError: name 'verticies' is not defined

graph1:36: NameError
   - [# 127] SVD graph1: 
--------------------------------------------------------------------------------
   35:         '''
   36:         if len(self.graph) > self.verticies:
   37:             raise IndexError('Too many verticies in graph.')
   38:         
-  39:         if source in self.graph:
+  39:         if source in graph:
   40:             self.graph[source].append(destination)
   41:         else:
   42:             self.graph[source] = [destination]
   43:         
--------------------------------------------------------------------------------
[0.19536 s] killed by ts-3-5/test_3-5_0-9_298.py::test_graph_add_edge

def test_graph_add_edge():
        g = Graph(4)
>       g.add_edge(0, 1)

ts-3-5/test_3-5_0-9_298.py:15: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph1.Graph object at 0x702648b4b760>, source = 0, destination = 1

>   ???
E   NameError: name 'graph' is not defined

graph1:39: NameError
   - [# 128] SVD graph1: 
--------------------------------------------------------------------------------
   36:         if len(self.graph) > self.verticies:
   37:             raise IndexError('Too many verticies in graph.')
   38:         
   39:         if source in self.graph:
-  40:             self.graph[source].append(destination)
+  40:             graph[source].append(destination)
   41:         else:
   42:             self.graph[source] = [destination]
   43:         
   44:         if destination not in self.graph:
--------------------------------------------------------------------------------
[0.19809 s] killed by ts-3-5/test_3-5_0-9_298.py::test_graph_has_cycle

def test_graph_has_cycle():
        g = Graph(4)
        g.add_edge(0, 1)
>       g.add_edge(1, 2)

ts-3-5/test_3-5_0-9_298.py:21: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph1.Graph object at 0x702648ec71f0>, source = 1, destination = 2

>   ???
E   NameError: name 'graph' is not defined

graph1:40: NameError
   - [# 129] SVD graph1: 
--------------------------------------------------------------------------------
   38:         
   39:         if source in self.graph:
   40:             self.graph[source].append(destination)
   41:         else:
-  42:             self.graph[source] = [destination]
+  42:             graph[source] = [destination]
   43:         
   44:         if destination not in self.graph:
   45:             self.graph[destination] = []
   46:     
--------------------------------------------------------------------------------
[0.20119 s] killed by ts-3-5/test_3-5_0-9_298.py::test_graph_add_edge

def test_graph_add_edge():
        g = Graph(4)
>       g.add_edge(0, 1)

ts-3-5/test_3-5_0-9_298.py:15: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph1.Graph object at 0x702648d6c730>, source = 0, destination = 1

>   ???
E   NameError: name 'graph' is not defined

graph1:42: NameError
   - [# 130] SVD graph1: 
--------------------------------------------------------------------------------
   40:             self.graph[source].append(destination)
   41:         else:
   42:             self.graph[source] = [destination]
   43:         
-  44:         if destination not in self.graph:
+  44:         if destination not in graph:
   45:             self.graph[destination] = []
   46:     
   47:     def has_cycle(self):
   48:         '''
--------------------------------------------------------------------------------
[0.17817 s] killed by ts-3-5/test_3-5_0-9_298.py::test_graph_add_edge

def test_graph_add_edge():
        g = Graph(4)
>       g.add_edge(0, 1)

ts-3-5/test_3-5_0-9_298.py:15: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph1.Graph object at 0x702648d5f310>, source = 0, destination = 1

>   ???
E   NameError: name 'graph' is not defined

graph1:44: NameError
   - [# 131] SVD graph1: 
--------------------------------------------------------------------------------
   41:         else:
   42:             self.graph[source] = [destination]
   43:         
   44:         if destination not in self.graph:
-  45:             self.graph[destination] = []
+  45:             graph[destination] = []
   46:     
   47:     def has_cycle(self):
   48:         '''
   49:         Detect if a graph has a cycle.
--------------------------------------------------------------------------------
[0.19336 s] killed by ts-3-5/test_3-5_0-9_298.py::test_graph_add_edge

def test_graph_add_edge():
        g = Graph(4)
>       g.add_edge(0, 1)

ts-3-5/test_3-5_0-9_298.py:15: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph1.Graph object at 0x702648a65810>, source = 0, destination = 1

>   ???
E   NameError: name 'graph' is not defined

graph1:45: NameError
   - [# 132] SVD graph1: 
--------------------------------------------------------------------------------
   53:             False if the graph is acyclic.
   54: 
   55:         '''
   56:         
-  57:         visited = [0] * self.verticies
+  57:         visited = [0] * verticies
   58:         
   59:         def valid(node):
   60:             
   61:             if visited[node] == -1:
--------------------------------------------------------------------------------
[0.20065 s] killed by ts-3-5/test_3-5_0-9_298.py::test_graph_has_cycle

def test_graph_has_cycle():
        g = Graph(4)
        g.add_edge(0, 1)
        g.add_edge(1, 2)
        g.add_edge(2, 0)
>       assert g.has_cycle() == True

ts-3-5/test_3-5_0-9_298.py:23: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph1.Graph object at 0x70264894e740>

>   ???
E   NameError: name 'verticies' is not defined

graph1:57: NameError
   - [# 133] SVD graph1: 
--------------------------------------------------------------------------------
   62:                 return False
   63:             elif visited[node] == 1:
   64:                 return True
   65:             visited[node] = -1
-  66:             for neighbor in self.graph[node]:
+  66:             for neighbor in graph[node]:
   67:                 if not (valid(neighbor)):
   68:                     return False
   69:             return True
   70:         
--------------------------------------------------------------------------------
[0.20014 s] killed by ts-3-5/test_3-5_0-9_298.py::test_graph_has_cycle

def test_graph_has_cycle():
        g = Graph(4)
        g.add_edge(0, 1)
        g.add_edge(1, 2)
        g.add_edge(2, 0)
>       assert g.has_cycle() == True

ts-3-5/test_3-5_0-9_298.py:23: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
graph1:77: in has_cycle
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

node = 0

>   ???
E   NameError: name 'graph' is not defined

graph1:66: NameError
   - [# 134] SVD graph1: 
--------------------------------------------------------------------------------
   67:                 if not (valid(neighbor)):
   68:                     return False
   69:             return True
   70:         
-  71:         for node in range(self.verticies):
+  71:         for node in range(verticies):
   72:             
   73:             
   74:             
   75:             if not (visited[node]):
--------------------------------------------------------------------------------
[0.20207 s] killed by ts-3-5/test_3-5_0-9_298.py::test_graph_has_cycle

def test_graph_has_cycle():
        g = Graph(4)
        g.add_edge(0, 1)
        g.add_edge(1, 2)
        g.add_edge(2, 0)
>       assert g.has_cycle() == True

ts-3-5/test_3-5_0-9_298.py:23: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph1.Graph object at 0x70264894f760>

>   ???
E   NameError: name 'verticies' is not defined

graph1:71: NameError
   - [# 135] SVD graph1: 
--------------------------------------------------------------------------------
  108:         stack = deque()
  109:         
  110:         def dfs(vertex):
  111:             visited.add(vertex)
- 112:             for j in self.graph[vertex]:
+ 112:             for j in graph[vertex]:
  113:                 if j not in visited:
  114:                     dfs(j)
  115:             
  116:             stack.appendleft(vertex)
--------------------------------------------------------------------------------
[0.18643 s] killed by ts-3-5/test_3-5_0-9_298.py::test_graph_topological_sort

def test_graph_topological_sort():
        g = Graph(4)
        g.add_edge(0, 1)
        g.add_edge(0, 2)
        g.add_edge(1, 3)
        g.add_edge(2, 3)
>       assert g.topological_sort() == [0, 2, 1, 3]

ts-3-5/test_3-5_0-9_298.py:31: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
graph1:120: in topological_sort
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

vertex = 0

>   ???
E   NameError: name 'graph' is not defined

graph1:112: NameError
   - [# 136] SVD graph1: 
--------------------------------------------------------------------------------
  114:                     dfs(j)
  115:             
  116:             stack.appendleft(vertex)
  117:         
- 118:         for (key, _) in self.graph.items():
+ 118:         for (key, _) in graph.items():
  119:             if key not in visited:
  120:                 dfs(key)
  121:         
  122:         return list(stack)
--------------------------------------------------------------------------------
[0.19868 s] killed by ts-3-5/test_3-5_0-9_298.py::test_graph_topological_sort

def test_graph_topological_sort():
        g = Graph(4)
        g.add_edge(0, 1)
        g.add_edge(0, 2)
        g.add_edge(1, 3)
        g.add_edge(2, 3)
>       assert g.topological_sort() == [0, 2, 1, 3]

ts-3-5/test_3-5_0-9_298.py:31: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph1.Graph object at 0x7026488fdf90>

>   ???
E   NameError: name 'graph' is not defined

graph1:118: NameError
   - [# 137] SVD graph1: 
--------------------------------------------------------------------------------
  123: 
  124: class WeightedGraphNode:
  125:     
  126:     def __init__(self, val):
- 127:         self.val = val
+ 127:         val = val
  128:         self.adjacent = {}
  129:     
  130:     def __str__(self):
  131:         return (str(self.val) + ' adjacent: ') + str([x for x in self.adjacent])
--------------------------------------------------------------------------------
[0.00000 s] incompetent

ValueError: expression must have Store context but has Load instead

   - [# 138] SVD graph1: 
--------------------------------------------------------------------------------
  124: class WeightedGraphNode:
  125:     
  126:     def __init__(self, val):
  127:         self.val = val
- 128:         self.adjacent = {}
+ 128:         adjacent = {}
  129:     
  130:     def __str__(self):
  131:         return (str(self.val) + ' adjacent: ') + str([x for x in self.adjacent])
  132:     
--------------------------------------------------------------------------------
[0.00000 s] incompetent

ValueError: expression must have Store context but has Load instead

   - [# 139] SVD graph1: 
--------------------------------------------------------------------------------
  127:         self.val = val
  128:         self.adjacent = {}
  129:     
  130:     def __str__(self):
- 131:         return (str(self.val) + ' adjacent: ') + str([x for x in self.adjacent])
+ 131:         return (str(val) + ' adjacent: ') + str([x for x in self.adjacent])
  132:     
  133:     def add_adjacent(self, node, weight=0):
  134:         self.adjacent[node] = weight
  135:     
--------------------------------------------------------------------------------
[0.17942 s] survived
   - [# 140] SVD graph1: 
--------------------------------------------------------------------------------
  127:         self.val = val
  128:         self.adjacent = {}
  129:     
  130:     def __str__(self):
- 131:         return (str(self.val) + ' adjacent: ') + str([x for x in self.adjacent])
+ 131:         return (str(self.val) + ' adjacent: ') + str([x for x in adjacent])
  132:     
  133:     def add_adjacent(self, node, weight=0):
  134:         self.adjacent[node] = weight
  135:     
--------------------------------------------------------------------------------
[0.17364 s] survived
   - [# 141] SVD graph1: 
--------------------------------------------------------------------------------
  130:     def __str__(self):
  131:         return (str(self.val) + ' adjacent: ') + str([x for x in self.adjacent])
  132:     
  133:     def add_adjacent(self, node, weight=0):
- 134:         self.adjacent[node] = weight
+ 134:         adjacent[node] = weight
  135:     
  136:     def remove_adjacent(self, node):
  137:         del self.adjacent[node]
  138: 
--------------------------------------------------------------------------------
[0.19861 s] killed by ts-3-5/test_3-5_0-9_298.py::test_weighted_graph_add_edge

def test_weighted_graph_add_edge():
        wg = WeightedGraph()
>       wg.add_edge(0, 1, 5)

ts-3-5/test_3-5_0-9_298.py:36: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
graph1:156: in add_edge
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph1.WeightedGraphNode object at 0x702648e3d630>, node = 1, weight = 5

>   ???
E   NameError: name 'adjacent' is not defined

graph1:134: NameError
   - [# 142] SVD graph1: 
--------------------------------------------------------------------------------
  133:     def add_adjacent(self, node, weight=0):
  134:         self.adjacent[node] = weight
  135:     
  136:     def remove_adjacent(self, node):
- 137:         del self.adjacent[node]
+ 137:         del adjacent[node]
  138: 
  139: class WeightedGraph:
  140:     
  141:     def __init__(self):
--------------------------------------------------------------------------------
[0.17989 s] killed by ts-3-5/test_3-5_0-9_298.py::test_weighted_graph_remove_edge

def test_weighted_graph_remove_edge():
        wg = WeightedGraph()
        wg.add_edge(0, 1, 5)
>       wg.remove_edge(0, 1)

ts-3-5/test_3-5_0-9_298.py:42: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
graph1:164: in remove_edge
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph1.WeightedGraphNode object at 0x702648bf5000>, node = 1

>   ???
E   NameError: name 'adjacent' is not defined

graph1:137: NameError
   - [# 143] SVD graph1: 
--------------------------------------------------------------------------------
  138: 
  139: class WeightedGraph:
  140:     
  141:     def __init__(self):
- 142:         self.vertices = {}
+ 142:         vertices = {}
  143:     
  144:     def __str__(self):
  145:         return ', '.join([node.__str__() for node in self.vertices.values()])
  146:     
--------------------------------------------------------------------------------
[0.00000 s] incompetent

ValueError: expression must have Store context but has Load instead

   - [# 144] SVD graph1: 
--------------------------------------------------------------------------------
  141:     def __init__(self):
  142:         self.vertices = {}
  143:     
  144:     def __str__(self):
- 145:         return ', '.join([node.__str__() for node in self.vertices.values()])
+ 145:         return ', '.join([node.__str__() for node in vertices.values()])
  146:     
  147:     def add_vertex(self, val):
  148:         vertex = WeightedGraphNode(val)
  149:         self.vertices[val] = vertex
--------------------------------------------------------------------------------
[0.17415 s] survived
   - [# 145] SVD graph1: 
--------------------------------------------------------------------------------
  145:         return ', '.join([node.__str__() for node in self.vertices.values()])
  146:     
  147:     def add_vertex(self, val):
  148:         vertex = WeightedGraphNode(val)
- 149:         self.vertices[val] = vertex
+ 149:         vertices[val] = vertex
  150:     
  151:     def add_edge(self, src, dst, weight=0):
  152:         if src not in self.vertices:
  153:             self.add_vertex(src)
--------------------------------------------------------------------------------
[0.21491 s] killed by ts-3-5/test_3-5_0-9_298.py::test_weighted_graph_add_edge

def test_weighted_graph_add_edge():
        wg = WeightedGraph()
>       wg.add_edge(0, 1, 5)

ts-3-5/test_3-5_0-9_298.py:36: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
graph1:153: in add_edge
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph1.WeightedGraph object at 0x702648a67430>, val = 0

>   ???
E   NameError: name 'vertices' is not defined

graph1:149: NameError
   - [# 146] SVD graph1: 
--------------------------------------------------------------------------------
  148:         vertex = WeightedGraphNode(val)
  149:         self.vertices[val] = vertex
  150:     
  151:     def add_edge(self, src, dst, weight=0):
- 152:         if src not in self.vertices:
+ 152:         if src not in vertices:
  153:             self.add_vertex(src)
  154:         if dst not in self.vertices:
  155:             self.add_vertex(dst)
  156:         self.vertices[src].add_adjacent(dst, weight)
--------------------------------------------------------------------------------
[0.19690 s] killed by ts-3-5/test_3-5_0-9_298.py::test_weighted_graph_add_edge

def test_weighted_graph_add_edge():
        wg = WeightedGraph()
>       wg.add_edge(0, 1, 5)

ts-3-5/test_3-5_0-9_298.py:36: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph1.WeightedGraph object at 0x702648ae58a0>, src = 0, dst = 1
weight = 5

>   ???
E   NameError: name 'vertices' is not defined

graph1:152: NameError
   - [# 147] SVD graph1: 
--------------------------------------------------------------------------------
  149:         self.vertices[val] = vertex
  150:     
  151:     def add_edge(self, src, dst, weight=0):
  152:         if src not in self.vertices:
- 153:             self.add_vertex(src)
+ 153:             add_vertex(src)
  154:         if dst not in self.vertices:
  155:             self.add_vertex(dst)
  156:         self.vertices[src].add_adjacent(dst, weight)
  157:     
--------------------------------------------------------------------------------
[0.22646 s] killed by ts-3-5/test_3-5_0-9_298.py::test_weighted_graph_add_edge

def test_weighted_graph_add_edge():
        wg = WeightedGraph()
>       wg.add_edge(0, 1, 5)

ts-3-5/test_3-5_0-9_298.py:36: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph1.WeightedGraph object at 0x702648bd57b0>, src = 0, dst = 1
weight = 5

>   ???
E   NameError: name 'add_vertex' is not defined

graph1:153: NameError
   - [# 148] SVD graph1: 
--------------------------------------------------------------------------------
  150:     
  151:     def add_edge(self, src, dst, weight=0):
  152:         if src not in self.vertices:
  153:             self.add_vertex(src)
- 154:         if dst not in self.vertices:
+ 154:         if dst not in vertices:
  155:             self.add_vertex(dst)
  156:         self.vertices[src].add_adjacent(dst, weight)
  157:     
  158:     def remove_edge(self, src, dst):
--------------------------------------------------------------------------------
[0.18205 s] killed by ts-3-5/test_3-5_0-9_298.py::test_weighted_graph_add_edge

def test_weighted_graph_add_edge():
        wg = WeightedGraph()
>       wg.add_edge(0, 1, 5)

ts-3-5/test_3-5_0-9_298.py:36: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph1.WeightedGraph object at 0x702648d9fb50>, src = 0, dst = 1
weight = 5

>   ???
E   NameError: name 'vertices' is not defined

graph1:154: NameError
   - [# 149] SVD graph1: 
--------------------------------------------------------------------------------
  151:     def add_edge(self, src, dst, weight=0):
  152:         if src not in self.vertices:
  153:             self.add_vertex(src)
  154:         if dst not in self.vertices:
- 155:             self.add_vertex(dst)
+ 155:             add_vertex(dst)
  156:         self.vertices[src].add_adjacent(dst, weight)
  157:     
  158:     def remove_edge(self, src, dst):
  159:         if src not in self.vertices:
--------------------------------------------------------------------------------
[0.19411 s] killed by ts-3-5/test_3-5_0-9_298.py::test_weighted_graph_add_edge

def test_weighted_graph_add_edge():
        wg = WeightedGraph()
>       wg.add_edge(0, 1, 5)

ts-3-5/test_3-5_0-9_298.py:36: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph1.WeightedGraph object at 0x702648e12fb0>, src = 0, dst = 1
weight = 5

>   ???
E   NameError: name 'add_vertex' is not defined

graph1:155: NameError
   - [# 150] SVD graph1: 
--------------------------------------------------------------------------------
  152:         if src not in self.vertices:
  153:             self.add_vertex(src)
  154:         if dst not in self.vertices:
  155:             self.add_vertex(dst)
- 156:         self.vertices[src].add_adjacent(dst, weight)
+ 156:         vertices[src].add_adjacent(dst, weight)
  157:     
  158:     def remove_edge(self, src, dst):
  159:         if src not in self.vertices:
  160:             return
--------------------------------------------------------------------------------
[0.19722 s] killed by ts-3-5/test_3-5_0-9_298.py::test_weighted_graph_add_edge

def test_weighted_graph_add_edge():
        wg = WeightedGraph()
>       wg.add_edge(0, 1, 5)

ts-3-5/test_3-5_0-9_298.py:36: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph1.WeightedGraph object at 0x702648a66e00>, src = 0, dst = 1
weight = 5

>   ???
E   NameError: name 'vertices' is not defined

graph1:156: NameError
   - [# 151] SVD graph1: 
--------------------------------------------------------------------------------
  155:             self.add_vertex(dst)
  156:         self.vertices[src].add_adjacent(dst, weight)
  157:     
  158:     def remove_edge(self, src, dst):
- 159:         if src not in self.vertices:
+ 159:         if src not in vertices:
  160:             return
  161:         if dst not in self.vertices:
  162:             return
  163:         if dst in self.vertices[src].adjacent:
--------------------------------------------------------------------------------
[0.20596 s] killed by ts-3-5/test_3-5_0-9_298.py::test_weighted_graph_remove_edge

def test_weighted_graph_remove_edge():
        wg = WeightedGraph()
        wg.add_edge(0, 1, 5)
>       wg.remove_edge(0, 1)

ts-3-5/test_3-5_0-9_298.py:42: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph1.WeightedGraph object at 0x702648d9dff0>, src = 0, dst = 1

>   ???
E   NameError: name 'vertices' is not defined

graph1:159: NameError
   - [# 152] SVD graph1: 
--------------------------------------------------------------------------------
  157:     
  158:     def remove_edge(self, src, dst):
  159:         if src not in self.vertices:
  160:             return
- 161:         if dst not in self.vertices:
+ 161:         if dst not in vertices:
  162:             return
  163:         if dst in self.vertices[src].adjacent:
  164:             self.vertices[src].remove_adjacent(dst)
--------------------------------------------------------------------------------
[0.19556 s] killed by ts-3-5/test_3-5_0-9_298.py::test_weighted_graph_remove_edge

def test_weighted_graph_remove_edge():
        wg = WeightedGraph()
        wg.add_edge(0, 1, 5)
>       wg.remove_edge(0, 1)

ts-3-5/test_3-5_0-9_298.py:42: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph1.WeightedGraph object at 0x702648d45c00>, src = 0, dst = 1

>   ???
E   NameError: name 'vertices' is not defined

graph1:161: NameError
   - [# 153] SVD graph1: 
--------------------------------------------------------------------------------
  159:         if src not in self.vertices:
  160:             return
  161:         if dst not in self.vertices:
  162:             return
- 163:         if dst in self.vertices[src].adjacent:
+ 163:         if dst in vertices[src].adjacent:
  164:             self.vertices[src].remove_adjacent(dst)
--------------------------------------------------------------------------------
[0.19621 s] killed by ts-3-5/test_3-5_0-9_298.py::test_weighted_graph_remove_edge

def test_weighted_graph_remove_edge():
        wg = WeightedGraph()
        wg.add_edge(0, 1, 5)
>       wg.remove_edge(0, 1)

ts-3-5/test_3-5_0-9_298.py:42: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph1.WeightedGraph object at 0x702648ddd7e0>, src = 0, dst = 1

>   ???
E   NameError: name 'vertices' is not defined

graph1:163: NameError
   - [# 154] SVD graph1: 
--------------------------------------------------------------------------------
  160:             return
  161:         if dst not in self.vertices:
  162:             return
  163:         if dst in self.vertices[src].adjacent:
- 164:             self.vertices[src].remove_adjacent(dst)
+ 164:             vertices[src].remove_adjacent(dst)
--------------------------------------------------------------------------------
[0.19592 s] killed by ts-3-5/test_3-5_0-9_298.py::test_weighted_graph_remove_edge

def test_weighted_graph_remove_edge():
        wg = WeightedGraph()
        wg.add_edge(0, 1, 5)
>       wg.remove_edge(0, 1)

ts-3-5/test_3-5_0-9_298.py:42: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph1.WeightedGraph object at 0x702648ae5ff0>, src = 0, dst = 1

>   ???
E   NameError: name 'vertices' is not defined

graph1:164: NameError
   - [# 155] ZIL graph1: 
--------------------------------------------------------------------------------
   63:             elif visited[node] == 1:
   64:                 return True
   65:             visited[node] = -1
   66:             for neighbor in self.graph[node]:
-  67:                 if not (valid(neighbor)):
-  68:                     return False
+  67:                 break
+  68:             
   69:             return True
   70:         
   71:         for node in range(self.verticies):
   72:             
--------------------------------------------------------------------------------
[0.21947 s] killed by ts-3-5/test_3-5_0-9_298.py::test_graph_has_cycle

def test_graph_has_cycle():
        g = Graph(4)
        g.add_edge(0, 1)
        g.add_edge(1, 2)
        g.add_edge(2, 0)
>       assert g.has_cycle() == True

ts-3-5/test_3-5_0-9_298.py:23: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
graph1:77: in has_cycle
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

node = 3

>   ???
E   KeyError: 3

graph1:66: KeyError
   - [# 156] ZIL graph1: 
--------------------------------------------------------------------------------
   71:         for node in range(self.verticies):
   72:             
   73:             
   74:             
-  75:             if not (visited[node]):
-  76:                 
-  77:                 if valid(node):
-  78:                     
-  79:                     visited = list(map((lambda x: (abs(x))), visited))
-  80:                 else:
-  81:                     
-  82:                     return True
+  75:             break
+  76:         
+  77:         
+  78:         
+  79:         
+  80:         
+  81:         
+  82:         
   83:         
   84:         return False
   85:     
   86:     def topological_sort(self):
--------------------------------------------------------------------------------
[0.20094 s] killed by ts-3-5/test_3-5_0-9_298.py::test_graph_has_cycle

def test_graph_has_cycle():
        g = Graph(4)
        g.add_edge(0, 1)
        g.add_edge(1, 2)
        g.add_edge(2, 0)
>       assert g.has_cycle() == True
E       AssertionError

ts-3-5/test_3-5_0-9_298.py:23: AssertionError
   - [# 157] ZIL graph1: 
--------------------------------------------------------------------------------
  109:         
  110:         def dfs(vertex):
  111:             visited.add(vertex)
  112:             for j in self.graph[vertex]:
- 113:                 if j not in visited:
- 114:                     dfs(j)
+ 113:                 break
+ 114:             
  115:             
  116:             stack.appendleft(vertex)
  117:         
  118:         for (key, _) in self.graph.items():
--------------------------------------------------------------------------------
[0.20405 s] killed by ts-3-5/test_3-5_0-9_298.py::test_graph_topological_sort

def test_graph_topological_sort():
        g = Graph(4)
        g.add_edge(0, 1)
        g.add_edge(0, 2)
        g.add_edge(1, 3)
        g.add_edge(2, 3)
>       assert g.topological_sort() == [0, 2, 1, 3]
E       AssertionError

ts-3-5/test_3-5_0-9_298.py:31: AssertionError
   - [# 158] ZIL graph1: 
--------------------------------------------------------------------------------
  115:             
  116:             stack.appendleft(vertex)
  117:         
  118:         for (key, _) in self.graph.items():
- 119:             if key not in visited:
- 120:                 dfs(key)
+ 119:             break
+ 120:         
  121:         
  122:         return list(stack)
  123: 
  124: class WeightedGraphNode:
--------------------------------------------------------------------------------
[0.21087 s] killed by ts-3-5/test_3-5_0-9_298.py::test_graph_topological_sort

def test_graph_topological_sort():
        g = Graph(4)
        g.add_edge(0, 1)
        g.add_edge(0, 2)
        g.add_edge(1, 3)
        g.add_edge(2, 3)
>       assert g.topological_sort() == [0, 2, 1, 3]
E       AssertionError

ts-3-5/test_3-5_0-9_298.py:31: AssertionError
[*] Mutation score [40.56873 s]: 71.9%
   - all: 158
   - killed: 100 (63.3%)
   - survived: 39 (24.7%)
   - incompetent: 19 (12.0%)
   - timeout: 0 (0.0%)
